{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "from scipy.stats.stats import pearsonr\n",
    "import pingouin as pg\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/apps/software/standard/core/anaconda/5.2.0-py3.6/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (13468,13469,13470) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "# Load RNA Seq Data\n",
    "rna_seq = pd.read_csv('/project/DSone/as3ek/data/csv_files/transcriptome_merged.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>PID</th>\n",
       "      <th>RNASes_Bx_loc</th>\n",
       "      <th>Diagnosis</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>A4GNT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>...</th>\n",
       "      <th>ZXDB</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "      <th>SEEM ID</th>\n",
       "      <th>Bx Date</th>\n",
       "      <th>Accession #</th>\n",
       "      <th>FileNames</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>2540002-9</td>\n",
       "      <td>C002</td>\n",
       "      <td>Doudenum</td>\n",
       "      <td>CELIAC</td>\n",
       "      <td>2.347019</td>\n",
       "      <td>25.421972</td>\n",
       "      <td>124.48038</td>\n",
       "      <td>1.980171</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>18.622360</td>\n",
       "      <td>...</td>\n",
       "      <td>1.375860</td>\n",
       "      <td>5.907832</td>\n",
       "      <td>3.748820</td>\n",
       "      <td>32.835430</td>\n",
       "      <td>22.541649</td>\n",
       "      <td>11.531942</td>\n",
       "      <td>C-002</td>\n",
       "      <td>2017-04-13 00:00:00</td>\n",
       "      <td>GI-17-3505A</td>\n",
       "      <td>GI-17-3505A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>2540004-9</td>\n",
       "      <td>C004</td>\n",
       "      <td>Doudenum</td>\n",
       "      <td>CELIAC</td>\n",
       "      <td>1.944927</td>\n",
       "      <td>27.633224</td>\n",
       "      <td>78.76862</td>\n",
       "      <td>1.473732</td>\n",
       "      <td>1.782160</td>\n",
       "      <td>13.545438</td>\n",
       "      <td>...</td>\n",
       "      <td>0.710877</td>\n",
       "      <td>3.942007</td>\n",
       "      <td>2.528200</td>\n",
       "      <td>25.566238</td>\n",
       "      <td>15.254350</td>\n",
       "      <td>6.072643</td>\n",
       "      <td>C-004</td>\n",
       "      <td>2017-05-01 00:00:00</td>\n",
       "      <td>GI-17-4060A</td>\n",
       "      <td>GI-17-4060A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>2540031-9</td>\n",
       "      <td>C031</td>\n",
       "      <td>Doudenum</td>\n",
       "      <td>CELIAC</td>\n",
       "      <td>1.286928</td>\n",
       "      <td>23.610662</td>\n",
       "      <td>138.25372</td>\n",
       "      <td>2.306936</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>13.497106</td>\n",
       "      <td>...</td>\n",
       "      <td>1.172150</td>\n",
       "      <td>4.238639</td>\n",
       "      <td>4.012072</td>\n",
       "      <td>25.441893</td>\n",
       "      <td>14.689348</td>\n",
       "      <td>9.913745</td>\n",
       "      <td>C-031</td>\n",
       "      <td>2018-08-02 00:00:00</td>\n",
       "      <td>GI-18-07450A</td>\n",
       "      <td>GI-18-07450A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>2540032-9</td>\n",
       "      <td>C032</td>\n",
       "      <td>Doudenum</td>\n",
       "      <td>CELIAC</td>\n",
       "      <td>1.229353</td>\n",
       "      <td>22.671690</td>\n",
       "      <td>133.90640</td>\n",
       "      <td>1.742326</td>\n",
       "      <td>10.360900</td>\n",
       "      <td>12.895801</td>\n",
       "      <td>...</td>\n",
       "      <td>1.648440</td>\n",
       "      <td>5.001438</td>\n",
       "      <td>4.950277</td>\n",
       "      <td>21.775680</td>\n",
       "      <td>16.350965</td>\n",
       "      <td>13.422350</td>\n",
       "      <td>C-032</td>\n",
       "      <td>2018-08-07 00:00:00</td>\n",
       "      <td>GI-18-07655A</td>\n",
       "      <td>GI-18-07655A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>2540033-9</td>\n",
       "      <td>C033</td>\n",
       "      <td>Doudenum</td>\n",
       "      <td>CELIAC</td>\n",
       "      <td>1.975784</td>\n",
       "      <td>22.597551</td>\n",
       "      <td>101.89975</td>\n",
       "      <td>1.012606</td>\n",
       "      <td>0.891341</td>\n",
       "      <td>10.065294</td>\n",
       "      <td>...</td>\n",
       "      <td>1.340800</td>\n",
       "      <td>3.710309</td>\n",
       "      <td>2.990033</td>\n",
       "      <td>21.703770</td>\n",
       "      <td>14.308658</td>\n",
       "      <td>9.205167</td>\n",
       "      <td>C-033</td>\n",
       "      <td>2018-08-10 00:00:00</td>\n",
       "      <td>GI-18-07807A</td>\n",
       "      <td>GI-18-07807A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 13472 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene   PID RNASes_Bx_loc Diagnosis      A1BG       A1CF        A2M  \\\n",
       "0  2540002-9  C002      Doudenum    CELIAC  2.347019  25.421972  124.48038   \n",
       "1  2540004-9  C004      Doudenum    CELIAC  1.944927  27.633224   78.76862   \n",
       "2  2540031-9  C031      Doudenum    CELIAC  1.286928  23.610662  138.25372   \n",
       "3  2540032-9  C032      Doudenum    CELIAC  1.229353  22.671690  133.90640   \n",
       "4  2540033-9  C033      Doudenum    CELIAC  1.975784  22.597551  101.89975   \n",
       "\n",
       "     A4GALT      A4GNT       AAAS  ...      ZXDB      ZXDC    ZYG11B  \\\n",
       "0  1.980171   0.010000  18.622360  ...  1.375860  5.907832  3.748820   \n",
       "1  1.473732   1.782160  13.545438  ...  0.710877  3.942007  2.528200   \n",
       "2  2.306936   0.010000  13.497106  ...  1.172150  4.238639  4.012072   \n",
       "3  1.742326  10.360900  12.895801  ...  1.648440  5.001438  4.950277   \n",
       "4  1.012606   0.891341  10.065294  ...  1.340800  3.710309  2.990033   \n",
       "\n",
       "         ZYX      ZZEF1       ZZZ3  SEEM ID              Bx Date  \\\n",
       "0  32.835430  22.541649  11.531942    C-002  2017-04-13 00:00:00   \n",
       "1  25.566238  15.254350   6.072643    C-004  2017-05-01 00:00:00   \n",
       "2  25.441893  14.689348   9.913745    C-031  2018-08-02 00:00:00   \n",
       "3  21.775680  16.350965  13.422350    C-032  2018-08-07 00:00:00   \n",
       "4  21.703770  14.308658   9.205167    C-033  2018-08-10 00:00:00   \n",
       "\n",
       "    Accession #     FileNames  \n",
       "0   GI-17-3505A   GI-17-3505A  \n",
       "1   GI-17-4060A   GI-17-4060A  \n",
       "2  GI-18-07450A  GI-18-07450A  \n",
       "3  GI-18-07655A  GI-18-07655A  \n",
       "4  GI-18-07807A  GI-18-07807A  \n",
       "\n",
       "[5 rows x 13472 columns]"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna_seq[rna_seq['Diagnosis'].isin(['CELIAC', 'CONTROL'])].reset_index(drop=True).head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load features from the CNN model\n",
    "train_feat_max = pd.read_csv('/project/DSone/as3ek/data/csv_files/features_celiac__normal.csv')\n",
    "valid_feat_max = pd.read_csv('/project/DSone/as3ek/data/csv_files/features_celiac__normal_valid.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load cleaned EE RNA Seq Data\n",
    "ee_rna_seq = pd.read_csv('/project/DSone/as3ek/data/csv_files/ee_trascript.csv')\n",
    "ee_rna_seq = ee_rna_seq.drop(['Unnamed: 0'], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/apps/software/standard/core/anaconda/5.2.0-py3.6/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (13468,13469,13470) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "# Load RNA Seq Data\n",
    "rna_seq = pd.read_csv('/project/DSone/as3ek/data/csv_files/transcriptome_merged.csv')\n",
    "\n",
    "# Drop unnecessary columns\n",
    "rna_seq = rna_seq.drop(['Gene', 'PID', 'RNASes_Bx_loc', 'SEEM ID', 'Bx Date', 'Accession #'], axis=1)\n",
    "\n",
    "# Filter all data to get only normal RNA Seq data\n",
    "normal_rna_seq = rna_seq[rna_seq['Diagnosis'] == 'CONTROL'].reset_index(drop=True)\n",
    "normal_rna_seq = normal_rna_seq.drop('Diagnosis', axis=1)\n",
    "\n",
    "# Filter all data to get only normal RNA Seq data\n",
    "celiac_rna_seq = rna_seq[rna_seq['Diagnosis'] == 'CELIAC'].reset_index(drop=True)\n",
    "celiac_rna_seq = celiac_rna_seq.drop('Diagnosis', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>A4GNT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAC</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>...</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDB</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "      <th>FileNames</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>2.347019</td>\n",
       "      <td>25.421972</td>\n",
       "      <td>124.48038</td>\n",
       "      <td>1.980171</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>18.622360</td>\n",
       "      <td>11.078631</td>\n",
       "      <td>124.08652</td>\n",
       "      <td>2.730297</td>\n",
       "      <td>17.697521</td>\n",
       "      <td>...</td>\n",
       "      <td>7.279240</td>\n",
       "      <td>8.349922</td>\n",
       "      <td>24.039724</td>\n",
       "      <td>1.375860</td>\n",
       "      <td>5.907832</td>\n",
       "      <td>3.748820</td>\n",
       "      <td>32.835430</td>\n",
       "      <td>22.541649</td>\n",
       "      <td>11.531942</td>\n",
       "      <td>GI-17-3505A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1.944927</td>\n",
       "      <td>27.633224</td>\n",
       "      <td>78.76862</td>\n",
       "      <td>1.473732</td>\n",
       "      <td>1.782160</td>\n",
       "      <td>13.545438</td>\n",
       "      <td>13.048814</td>\n",
       "      <td>257.55106</td>\n",
       "      <td>1.789949</td>\n",
       "      <td>27.142975</td>\n",
       "      <td>...</td>\n",
       "      <td>5.700451</td>\n",
       "      <td>6.408322</td>\n",
       "      <td>24.005035</td>\n",
       "      <td>0.710877</td>\n",
       "      <td>3.942007</td>\n",
       "      <td>2.528200</td>\n",
       "      <td>25.566238</td>\n",
       "      <td>15.254350</td>\n",
       "      <td>6.072643</td>\n",
       "      <td>GI-17-4060A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1.286928</td>\n",
       "      <td>23.610662</td>\n",
       "      <td>138.25372</td>\n",
       "      <td>2.306936</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>13.497106</td>\n",
       "      <td>12.191676</td>\n",
       "      <td>101.12962</td>\n",
       "      <td>2.320212</td>\n",
       "      <td>24.512012</td>\n",
       "      <td>...</td>\n",
       "      <td>8.955390</td>\n",
       "      <td>10.433591</td>\n",
       "      <td>23.415869</td>\n",
       "      <td>1.172150</td>\n",
       "      <td>4.238639</td>\n",
       "      <td>4.012072</td>\n",
       "      <td>25.441893</td>\n",
       "      <td>14.689348</td>\n",
       "      <td>9.913745</td>\n",
       "      <td>GI-18-07450A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1.229353</td>\n",
       "      <td>22.671690</td>\n",
       "      <td>133.90640</td>\n",
       "      <td>1.742326</td>\n",
       "      <td>10.360900</td>\n",
       "      <td>12.895801</td>\n",
       "      <td>13.162550</td>\n",
       "      <td>276.89230</td>\n",
       "      <td>2.748407</td>\n",
       "      <td>20.668043</td>\n",
       "      <td>...</td>\n",
       "      <td>9.474550</td>\n",
       "      <td>7.819117</td>\n",
       "      <td>19.242214</td>\n",
       "      <td>1.648440</td>\n",
       "      <td>5.001438</td>\n",
       "      <td>4.950277</td>\n",
       "      <td>21.775680</td>\n",
       "      <td>16.350965</td>\n",
       "      <td>13.422350</td>\n",
       "      <td>GI-18-07655A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1.975784</td>\n",
       "      <td>22.597551</td>\n",
       "      <td>101.89975</td>\n",
       "      <td>1.012606</td>\n",
       "      <td>0.891341</td>\n",
       "      <td>10.065294</td>\n",
       "      <td>8.812082</td>\n",
       "      <td>169.54845</td>\n",
       "      <td>2.241079</td>\n",
       "      <td>16.227331</td>\n",
       "      <td>...</td>\n",
       "      <td>5.545079</td>\n",
       "      <td>5.369827</td>\n",
       "      <td>11.697079</td>\n",
       "      <td>1.340800</td>\n",
       "      <td>3.710309</td>\n",
       "      <td>2.990033</td>\n",
       "      <td>21.703770</td>\n",
       "      <td>14.308658</td>\n",
       "      <td>9.205167</td>\n",
       "      <td>GI-18-07807A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 13465 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       A1BG       A1CF        A2M    A4GALT      A4GNT       AAAS       AACS  \\\n",
       "0  2.347019  25.421972  124.48038  1.980171   0.010000  18.622360  11.078631   \n",
       "1  1.944927  27.633224   78.76862  1.473732   1.782160  13.545438  13.048814   \n",
       "2  1.286928  23.610662  138.25372  2.306936   0.010000  13.497106  12.191676   \n",
       "3  1.229353  22.671690  133.90640  1.742326  10.360900  12.895801  13.162550   \n",
       "4  1.975784  22.597551  101.89975  1.012606   0.891341  10.065294   8.812082   \n",
       "\n",
       "       AADAC     AAED1      AAGAB  ...      ZW10     ZWILCH      ZWINT  \\\n",
       "0  124.08652  2.730297  17.697521  ...  7.279240   8.349922  24.039724   \n",
       "1  257.55106  1.789949  27.142975  ...  5.700451   6.408322  24.005035   \n",
       "2  101.12962  2.320212  24.512012  ...  8.955390  10.433591  23.415869   \n",
       "3  276.89230  2.748407  20.668043  ...  9.474550   7.819117  19.242214   \n",
       "4  169.54845  2.241079  16.227331  ...  5.545079   5.369827  11.697079   \n",
       "\n",
       "       ZXDB      ZXDC    ZYG11B        ZYX      ZZEF1       ZZZ3     FileNames  \n",
       "0  1.375860  5.907832  3.748820  32.835430  22.541649  11.531942   GI-17-3505A  \n",
       "1  0.710877  3.942007  2.528200  25.566238  15.254350   6.072643   GI-17-4060A  \n",
       "2  1.172150  4.238639  4.012072  25.441893  14.689348   9.913745  GI-18-07450A  \n",
       "3  1.648440  5.001438  4.950277  21.775680  16.350965  13.422350  GI-18-07655A  \n",
       "4  1.340800  3.710309  2.990033  21.703770  14.308658   9.205167  GI-18-07807A  \n",
       "\n",
       "[5 rows x 13465 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "celiac_rna_seq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>A4GNT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAC</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>...</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDB</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "      <th>FileNames</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1.527489</td>\n",
       "      <td>39.161606</td>\n",
       "      <td>141.393680</td>\n",
       "      <td>2.250741</td>\n",
       "      <td>3.340980</td>\n",
       "      <td>13.393648</td>\n",
       "      <td>11.521561</td>\n",
       "      <td>190.26294</td>\n",
       "      <td>3.802879</td>\n",
       "      <td>27.817423</td>\n",
       "      <td>...</td>\n",
       "      <td>7.966211</td>\n",
       "      <td>5.824094</td>\n",
       "      <td>15.380531</td>\n",
       "      <td>1.90331</td>\n",
       "      <td>6.084198</td>\n",
       "      <td>5.986877</td>\n",
       "      <td>19.513960</td>\n",
       "      <td>30.423355</td>\n",
       "      <td>13.065509</td>\n",
       "      <td>GI-17-2781A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1.836630</td>\n",
       "      <td>20.815561</td>\n",
       "      <td>70.397380</td>\n",
       "      <td>1.366245</td>\n",
       "      <td>8.056910</td>\n",
       "      <td>13.729294</td>\n",
       "      <td>11.792672</td>\n",
       "      <td>188.86170</td>\n",
       "      <td>2.597826</td>\n",
       "      <td>19.777740</td>\n",
       "      <td>...</td>\n",
       "      <td>6.067741</td>\n",
       "      <td>3.992341</td>\n",
       "      <td>12.340939</td>\n",
       "      <td>1.04346</td>\n",
       "      <td>4.722662</td>\n",
       "      <td>2.653590</td>\n",
       "      <td>24.902603</td>\n",
       "      <td>14.547653</td>\n",
       "      <td>10.382214</td>\n",
       "      <td>GI-17-5415A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>2.070108</td>\n",
       "      <td>30.049267</td>\n",
       "      <td>78.129480</td>\n",
       "      <td>2.200778</td>\n",
       "      <td>8.668561</td>\n",
       "      <td>14.197790</td>\n",
       "      <td>13.160649</td>\n",
       "      <td>299.86722</td>\n",
       "      <td>2.576309</td>\n",
       "      <td>22.687897</td>\n",
       "      <td>...</td>\n",
       "      <td>6.487394</td>\n",
       "      <td>4.760917</td>\n",
       "      <td>12.637662</td>\n",
       "      <td>1.09973</td>\n",
       "      <td>5.131794</td>\n",
       "      <td>4.623030</td>\n",
       "      <td>23.566916</td>\n",
       "      <td>17.086746</td>\n",
       "      <td>8.982910</td>\n",
       "      <td>GI-17-5671A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>2.323137</td>\n",
       "      <td>27.274525</td>\n",
       "      <td>93.843880</td>\n",
       "      <td>2.508640</td>\n",
       "      <td>1.156290</td>\n",
       "      <td>15.076123</td>\n",
       "      <td>11.050514</td>\n",
       "      <td>176.36838</td>\n",
       "      <td>1.392020</td>\n",
       "      <td>19.399889</td>\n",
       "      <td>...</td>\n",
       "      <td>4.988570</td>\n",
       "      <td>2.749776</td>\n",
       "      <td>13.213606</td>\n",
       "      <td>1.35546</td>\n",
       "      <td>5.087205</td>\n",
       "      <td>2.442820</td>\n",
       "      <td>26.665813</td>\n",
       "      <td>20.346804</td>\n",
       "      <td>7.571774</td>\n",
       "      <td>GI-17-6710A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1.550304</td>\n",
       "      <td>28.912386</td>\n",
       "      <td>78.544624</td>\n",
       "      <td>2.113191</td>\n",
       "      <td>0.260323</td>\n",
       "      <td>14.713420</td>\n",
       "      <td>10.754949</td>\n",
       "      <td>272.44060</td>\n",
       "      <td>2.326984</td>\n",
       "      <td>22.267065</td>\n",
       "      <td>...</td>\n",
       "      <td>7.437720</td>\n",
       "      <td>5.820460</td>\n",
       "      <td>18.061235</td>\n",
       "      <td>1.39190</td>\n",
       "      <td>5.067470</td>\n",
       "      <td>3.432585</td>\n",
       "      <td>24.128992</td>\n",
       "      <td>15.404562</td>\n",
       "      <td>10.151729</td>\n",
       "      <td>GI-17-6728A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 13465 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       A1BG       A1CF         A2M    A4GALT     A4GNT       AAAS       AACS  \\\n",
       "0  1.527489  39.161606  141.393680  2.250741  3.340980  13.393648  11.521561   \n",
       "1  1.836630  20.815561   70.397380  1.366245  8.056910  13.729294  11.792672   \n",
       "2  2.070108  30.049267   78.129480  2.200778  8.668561  14.197790  13.160649   \n",
       "3  2.323137  27.274525   93.843880  2.508640  1.156290  15.076123  11.050514   \n",
       "4  1.550304  28.912386   78.544624  2.113191  0.260323  14.713420  10.754949   \n",
       "\n",
       "       AADAC     AAED1      AAGAB  ...      ZW10    ZWILCH      ZWINT  \\\n",
       "0  190.26294  3.802879  27.817423  ...  7.966211  5.824094  15.380531   \n",
       "1  188.86170  2.597826  19.777740  ...  6.067741  3.992341  12.340939   \n",
       "2  299.86722  2.576309  22.687897  ...  6.487394  4.760917  12.637662   \n",
       "3  176.36838  1.392020  19.399889  ...  4.988570  2.749776  13.213606   \n",
       "4  272.44060  2.326984  22.267065  ...  7.437720  5.820460  18.061235   \n",
       "\n",
       "      ZXDB      ZXDC    ZYG11B        ZYX      ZZEF1       ZZZ3    FileNames  \n",
       "0  1.90331  6.084198  5.986877  19.513960  30.423355  13.065509  GI-17-2781A  \n",
       "1  1.04346  4.722662  2.653590  24.902603  14.547653  10.382214  GI-17-5415A  \n",
       "2  1.09973  5.131794  4.623030  23.566916  17.086746   8.982910  GI-17-5671A  \n",
       "3  1.35546  5.087205  2.442820  26.665813  20.346804   7.571774  GI-17-6710A  \n",
       "4  1.39190  5.067470  3.432585  24.128992  15.404562  10.151729  GI-17-6728A  \n",
       "\n",
       "[5 rows x 13465 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_rna_seq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "name_map = pd.read_csv('/scratch/as3ek/histo_visual_recog/scripts/data/cchmc_mapping.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "feat_max = pd.concat([train_feat_max, valid_feat_max])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "gi_feat_max = feat_max[feat_max['fname'].str.contains('GI')].reset_index(drop=True)\n",
    "gi_feat_max['fname'] = gi_feat_max['fname'].str.split('__').str[0].str.split('_').str[0].str.replace('I', 'I-').str.replace(' ', '').str.replace('A1', 'A')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "c_feat_max = feat_max[feat_max['fname'].str.contains('C')].reset_index(drop=True)\n",
    "c_feat_max['fname'] = c_feat_max['fname'].str.split('__').str[0].str.split('_').str[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Group features for all patches by biopsy name\n",
    "gi_feat_max_grpd = gi_feat_max.groupby('fname', as_index=False).mean()\n",
    "c_feat_max_grpd = c_feat_max.groupby('fname', as_index=False).mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fname</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>...</th>\n",
       "      <th>2038</th>\n",
       "      <th>2039</th>\n",
       "      <th>2040</th>\n",
       "      <th>2041</th>\n",
       "      <th>2042</th>\n",
       "      <th>2043</th>\n",
       "      <th>2044</th>\n",
       "      <th>2045</th>\n",
       "      <th>2046</th>\n",
       "      <th>2047</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>C027</td>\n",
       "      <td>1.569540</td>\n",
       "      <td>1.800629</td>\n",
       "      <td>2.777871</td>\n",
       "      <td>1.160911</td>\n",
       "      <td>2.100492</td>\n",
       "      <td>1.973405</td>\n",
       "      <td>2.890815</td>\n",
       "      <td>3.700204</td>\n",
       "      <td>1.435636</td>\n",
       "      <td>...</td>\n",
       "      <td>2.146676</td>\n",
       "      <td>2.234218</td>\n",
       "      <td>1.676906</td>\n",
       "      <td>2.773962</td>\n",
       "      <td>1.852900</td>\n",
       "      <td>0.716745</td>\n",
       "      <td>1.483975</td>\n",
       "      <td>1.156939</td>\n",
       "      <td>1.616572</td>\n",
       "      <td>1.182015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>C031</td>\n",
       "      <td>2.226476</td>\n",
       "      <td>2.614796</td>\n",
       "      <td>1.191848</td>\n",
       "      <td>2.356947</td>\n",
       "      <td>1.677903</td>\n",
       "      <td>1.381091</td>\n",
       "      <td>1.735029</td>\n",
       "      <td>2.460516</td>\n",
       "      <td>3.493864</td>\n",
       "      <td>...</td>\n",
       "      <td>2.044576</td>\n",
       "      <td>1.216872</td>\n",
       "      <td>1.455889</td>\n",
       "      <td>0.954203</td>\n",
       "      <td>3.254052</td>\n",
       "      <td>1.042613</td>\n",
       "      <td>2.279874</td>\n",
       "      <td>1.648835</td>\n",
       "      <td>0.992039</td>\n",
       "      <td>1.662386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>C032</td>\n",
       "      <td>2.028006</td>\n",
       "      <td>2.937899</td>\n",
       "      <td>1.173732</td>\n",
       "      <td>2.243904</td>\n",
       "      <td>1.690223</td>\n",
       "      <td>1.474388</td>\n",
       "      <td>1.781492</td>\n",
       "      <td>2.154859</td>\n",
       "      <td>2.999383</td>\n",
       "      <td>...</td>\n",
       "      <td>2.040479</td>\n",
       "      <td>1.030623</td>\n",
       "      <td>1.442668</td>\n",
       "      <td>1.137877</td>\n",
       "      <td>3.027513</td>\n",
       "      <td>1.056706</td>\n",
       "      <td>2.569509</td>\n",
       "      <td>1.531834</td>\n",
       "      <td>0.929448</td>\n",
       "      <td>1.406503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>C033</td>\n",
       "      <td>1.950046</td>\n",
       "      <td>3.059400</td>\n",
       "      <td>1.209982</td>\n",
       "      <td>2.162022</td>\n",
       "      <td>1.867939</td>\n",
       "      <td>2.222352</td>\n",
       "      <td>1.583083</td>\n",
       "      <td>1.621948</td>\n",
       "      <td>2.341786</td>\n",
       "      <td>...</td>\n",
       "      <td>1.905583</td>\n",
       "      <td>1.521226</td>\n",
       "      <td>1.300404</td>\n",
       "      <td>1.074216</td>\n",
       "      <td>2.650147</td>\n",
       "      <td>0.974185</td>\n",
       "      <td>2.618763</td>\n",
       "      <td>1.584212</td>\n",
       "      <td>1.017927</td>\n",
       "      <td>1.649350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>C034</td>\n",
       "      <td>1.932541</td>\n",
       "      <td>3.410113</td>\n",
       "      <td>1.298151</td>\n",
       "      <td>2.051920</td>\n",
       "      <td>1.945213</td>\n",
       "      <td>1.997006</td>\n",
       "      <td>1.859203</td>\n",
       "      <td>2.207159</td>\n",
       "      <td>2.698378</td>\n",
       "      <td>...</td>\n",
       "      <td>2.181139</td>\n",
       "      <td>1.488641</td>\n",
       "      <td>1.476682</td>\n",
       "      <td>1.314644</td>\n",
       "      <td>2.687068</td>\n",
       "      <td>0.930547</td>\n",
       "      <td>2.500947</td>\n",
       "      <td>1.662968</td>\n",
       "      <td>1.039017</td>\n",
       "      <td>1.537389</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 2049 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  fname         0         1         2         3         4         5         6  \\\n",
       "0  C027  1.569540  1.800629  2.777871  1.160911  2.100492  1.973405  2.890815   \n",
       "1  C031  2.226476  2.614796  1.191848  2.356947  1.677903  1.381091  1.735029   \n",
       "2  C032  2.028006  2.937899  1.173732  2.243904  1.690223  1.474388  1.781492   \n",
       "3  C033  1.950046  3.059400  1.209982  2.162022  1.867939  2.222352  1.583083   \n",
       "4  C034  1.932541  3.410113  1.298151  2.051920  1.945213  1.997006  1.859203   \n",
       "\n",
       "          7         8  ...      2038      2039      2040      2041      2042  \\\n",
       "0  3.700204  1.435636  ...  2.146676  2.234218  1.676906  2.773962  1.852900   \n",
       "1  2.460516  3.493864  ...  2.044576  1.216872  1.455889  0.954203  3.254052   \n",
       "2  2.154859  2.999383  ...  2.040479  1.030623  1.442668  1.137877  3.027513   \n",
       "3  1.621948  2.341786  ...  1.905583  1.521226  1.300404  1.074216  2.650147   \n",
       "4  2.207159  2.698378  ...  2.181139  1.488641  1.476682  1.314644  2.687068   \n",
       "\n",
       "       2043      2044      2045      2046      2047  \n",
       "0  0.716745  1.483975  1.156939  1.616572  1.182015  \n",
       "1  1.042613  2.279874  1.648835  0.992039  1.662386  \n",
       "2  1.056706  2.569509  1.531834  0.929448  1.406503  \n",
       "3  0.974185  2.618763  1.584212  1.017927  1.649350  \n",
       "4  0.930547  2.500947  1.662968  1.039017  1.537389  \n",
       "\n",
       "[5 rows x 2049 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c_feat_max_grpd.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fname</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>...</th>\n",
       "      <th>2038</th>\n",
       "      <th>2039</th>\n",
       "      <th>2040</th>\n",
       "      <th>2041</th>\n",
       "      <th>2042</th>\n",
       "      <th>2043</th>\n",
       "      <th>2044</th>\n",
       "      <th>2045</th>\n",
       "      <th>2046</th>\n",
       "      <th>2047</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>GI-17-10289A</td>\n",
       "      <td>1.473160</td>\n",
       "      <td>1.664941</td>\n",
       "      <td>2.161360</td>\n",
       "      <td>1.368585</td>\n",
       "      <td>1.817131</td>\n",
       "      <td>2.512023</td>\n",
       "      <td>1.800048</td>\n",
       "      <td>2.280767</td>\n",
       "      <td>1.295154</td>\n",
       "      <td>...</td>\n",
       "      <td>1.824156</td>\n",
       "      <td>3.185858</td>\n",
       "      <td>2.918280</td>\n",
       "      <td>1.852221</td>\n",
       "      <td>0.787371</td>\n",
       "      <td>0.480029</td>\n",
       "      <td>0.747601</td>\n",
       "      <td>1.087269</td>\n",
       "      <td>3.073834</td>\n",
       "      <td>0.817811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>GI-17-10498A</td>\n",
       "      <td>1.917369</td>\n",
       "      <td>1.478766</td>\n",
       "      <td>1.801828</td>\n",
       "      <td>1.980441</td>\n",
       "      <td>1.442035</td>\n",
       "      <td>3.487439</td>\n",
       "      <td>1.396218</td>\n",
       "      <td>3.322703</td>\n",
       "      <td>1.369927</td>\n",
       "      <td>...</td>\n",
       "      <td>2.312621</td>\n",
       "      <td>3.988256</td>\n",
       "      <td>5.212091</td>\n",
       "      <td>1.797359</td>\n",
       "      <td>0.280776</td>\n",
       "      <td>0.429408</td>\n",
       "      <td>0.609097</td>\n",
       "      <td>1.044054</td>\n",
       "      <td>5.870791</td>\n",
       "      <td>0.747502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>GI-17-10706A</td>\n",
       "      <td>1.687944</td>\n",
       "      <td>1.440563</td>\n",
       "      <td>2.734079</td>\n",
       "      <td>1.206448</td>\n",
       "      <td>1.704757</td>\n",
       "      <td>2.729145</td>\n",
       "      <td>1.734858</td>\n",
       "      <td>2.927566</td>\n",
       "      <td>1.081435</td>\n",
       "      <td>...</td>\n",
       "      <td>2.002050</td>\n",
       "      <td>3.381622</td>\n",
       "      <td>3.009862</td>\n",
       "      <td>1.818307</td>\n",
       "      <td>0.813056</td>\n",
       "      <td>0.562646</td>\n",
       "      <td>0.768400</td>\n",
       "      <td>1.185493</td>\n",
       "      <td>3.329921</td>\n",
       "      <td>0.846993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>GI-17-2781A</td>\n",
       "      <td>1.495067</td>\n",
       "      <td>1.204786</td>\n",
       "      <td>2.941421</td>\n",
       "      <td>1.239465</td>\n",
       "      <td>1.391773</td>\n",
       "      <td>3.086098</td>\n",
       "      <td>1.855882</td>\n",
       "      <td>3.591309</td>\n",
       "      <td>1.278847</td>\n",
       "      <td>...</td>\n",
       "      <td>2.027716</td>\n",
       "      <td>3.655886</td>\n",
       "      <td>4.059050</td>\n",
       "      <td>2.225151</td>\n",
       "      <td>0.504351</td>\n",
       "      <td>0.408825</td>\n",
       "      <td>0.506742</td>\n",
       "      <td>0.872373</td>\n",
       "      <td>4.237109</td>\n",
       "      <td>1.067718</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>GI-17-3505A</td>\n",
       "      <td>1.744918</td>\n",
       "      <td>1.847216</td>\n",
       "      <td>1.341570</td>\n",
       "      <td>1.445397</td>\n",
       "      <td>2.693716</td>\n",
       "      <td>1.355879</td>\n",
       "      <td>1.372638</td>\n",
       "      <td>2.060323</td>\n",
       "      <td>1.648659</td>\n",
       "      <td>...</td>\n",
       "      <td>2.519777</td>\n",
       "      <td>2.549064</td>\n",
       "      <td>2.463550</td>\n",
       "      <td>0.919064</td>\n",
       "      <td>0.823797</td>\n",
       "      <td>0.913631</td>\n",
       "      <td>0.483297</td>\n",
       "      <td>1.921564</td>\n",
       "      <td>2.159932</td>\n",
       "      <td>1.290757</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 2049 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          fname         0         1         2         3         4         5  \\\n",
       "0  GI-17-10289A  1.473160  1.664941  2.161360  1.368585  1.817131  2.512023   \n",
       "1  GI-17-10498A  1.917369  1.478766  1.801828  1.980441  1.442035  3.487439   \n",
       "2  GI-17-10706A  1.687944  1.440563  2.734079  1.206448  1.704757  2.729145   \n",
       "3   GI-17-2781A  1.495067  1.204786  2.941421  1.239465  1.391773  3.086098   \n",
       "4   GI-17-3505A  1.744918  1.847216  1.341570  1.445397  2.693716  1.355879   \n",
       "\n",
       "          6         7         8  ...      2038      2039      2040      2041  \\\n",
       "0  1.800048  2.280767  1.295154  ...  1.824156  3.185858  2.918280  1.852221   \n",
       "1  1.396218  3.322703  1.369927  ...  2.312621  3.988256  5.212091  1.797359   \n",
       "2  1.734858  2.927566  1.081435  ...  2.002050  3.381622  3.009862  1.818307   \n",
       "3  1.855882  3.591309  1.278847  ...  2.027716  3.655886  4.059050  2.225151   \n",
       "4  1.372638  2.060323  1.648659  ...  2.519777  2.549064  2.463550  0.919064   \n",
       "\n",
       "       2042      2043      2044      2045      2046      2047  \n",
       "0  0.787371  0.480029  0.747601  1.087269  3.073834  0.817811  \n",
       "1  0.280776  0.429408  0.609097  1.044054  5.870791  0.747502  \n",
       "2  0.813056  0.562646  0.768400  1.185493  3.329921  0.846993  \n",
       "3  0.504351  0.408825  0.506742  0.872373  4.237109  1.067718  \n",
       "4  0.823797  0.913631  0.483297  1.921564  2.159932  1.290757  \n",
       "\n",
       "[5 rows x 2049 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gi_feat_max_grpd.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "name_map = name_map[['PID', 'Accession #']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge seq with features\n",
    "c_feat_max_grpd = pd.merge(c_feat_max_grpd, name_map, left_on='fname', right_on='PID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "c_feat_max_grpd = c_feat_max_grpd.drop(['fname', 'PID'], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "c_feat_max_grpd = c_feat_max_grpd.rename(columns={'Accession #':'fname'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "feat_max_grpd = pd.concat([gi_feat_max_grpd, c_feat_max_grpd], ignore_index=True, sort=False)\n",
    "feat_max_grpd.to_csv('/scratch/as3ek/histo_visual_recog/results/image_features_celiac_normal_clean.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "rna_seq = pd.concat([normal_rna_seq, celiac_rna_seq], ignore_index=True, sort=False)\n",
    "rna_seq.to_csv('/scratch/as3ek/histo_visual_recog/results/rna_seq_celiac_normal_clean.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge seq with features\n",
    "seq_feat = pd.merge(rna_seq, feat_max_grpd, left_on='FileNames', right_on='fname')\n",
    "seq_feat.to_csv('/scratch/as3ek/histo_visual_recog/results/rnaseq_feat_celiac_normal.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Getting list of all columns for correlation b/w\n",
    "x = list(rna_seq)\n",
    "x.remove('FileNames')\n",
    "\n",
    "y = list(feat_max_grpd)\n",
    "y.remove('fname')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Calculate pairwise correlation and filter it for reqd columns and round the values\n",
    "corr = seq_feat.corr()\n",
    "corr_fltrd = corr[x][corr.index.isin(y)]\n",
    "corr_fltrd = corr_fltrd.round(4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_fltrd.to_csv('/scratch/as3ek/histo_visual_recog/results/gene_feat_corr_celiac_normal.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_map = pd.read_csv('/scratch/as3ek/histo_visual_recog/scripts/data/custom.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "def corr_cutoff(corr_fltrd, cutoff):\n",
    "    # Make copy\n",
    "    df = corr_fltrd.copy().T\n",
    "    \n",
    "    # Replace all values below cutoff with 0\n",
    "    for col in y:\n",
    "        df.loc[np.abs(df[col]) < cutoff, col] = 0\n",
    "    \n",
    "    # Remove all columns with all zeros (Remove unnecessary features)\n",
    "    df = df.loc[:, (df != 0).any(axis=0)]\n",
    "    \n",
    "    # Transpose and remove all columns with all zeros (Remove unnecessary genes)\n",
    "    df = df.T\n",
    "    df = df.loc[:, (df != 0).any(axis=0)]\n",
    "    \n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "def compute_corr_metrics(corr_fltrd_cutoff, seq_feat, gene_map):\n",
    "    corr_metrics = pd.DataFrame(columns=['Gene Symbol', 'Gene Name', 'Feature', 'PearsonR', 'p-value'])\n",
    "    c = corr_fltrd_cutoff.copy()\n",
    "    for gene in c.columns:\n",
    "        df = pd.DataFrame(c[gene]).T\n",
    "        df = df.loc[:, (df != 0).any(axis=0)]\n",
    "        \n",
    "        names = list(gene_map[gene_map['Approved symbol'] == gene]['Approved name'])\n",
    "        if len(names) > 0:\n",
    "            name = names[0]\n",
    "        else:\n",
    "            name = gene\n",
    "        \n",
    "        for feat in df.columns:\n",
    "            r, p = pearsonr(seq_feat[feat], seq_feat[gene])\n",
    "            tmp = pd.DataFrame([[gene, name, feat, r, p]], columns=['Gene Symbol', 'Gene Name', 'Feature', 'PearsonR', 'p-value'])\n",
    "            corr_metrics = pd.concat([corr_metrics, tmp], ignore_index=True)\n",
    "    return corr_metrics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "def compute_regulation(normal_rna_seq, ee_rna_seq, genes, cutoff_fold, gene_map):\n",
    "    regulation_data = pd.DataFrame(columns=['Regulation', 'Gene Symbol', 'Gene Name', 'fold'])\n",
    "    for col in genes:\n",
    "        a = np.mean(ee_rna_seq[col]) / np.mean(normal_rna_seq[col]) \n",
    "        b = 1 / a\n",
    "        names = list(gene_map[gene_map['Approved symbol'] == col]['Approved name'])\n",
    "        if len(names) > 0:\n",
    "            name = names[0]\n",
    "        else:\n",
    "            name = col\n",
    "        if np.abs(a) > cutoff_fold:\n",
    "            tmp = pd.DataFrame([['up', col, name, a]], columns=['Regulation', 'Gene Symbol', 'Gene Name', 'fold'])\n",
    "            regulation_data = pd.concat([regulation_data, tmp], ignore_index=True)\n",
    "        if np.abs(b) > cutoff_fold:\n",
    "            tmp = pd.DataFrame([['down', col, name, b]], columns=['Regulation', 'Gene Symbol', 'Gene Name', 'fold'])\n",
    "            regulation_data = pd.concat([regulation_data, tmp], ignore_index=True)\n",
    "    \n",
    "    return regulation_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "regulation_data = compute_regulation(normal_rna_seq, ee_rna_seq, x, 1, gene_map)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "regulation_data.to_csv('/scratch/as3ek/histo_visual_recog/scripts/data/gene_regulation_data_celiac_normal.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "corr_fltrd_cutoff = corr_cutoff(corr_fltrd, 0.7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "plt.figure(figsize=(30, 200))\n",
    "sns.heatmap(corr_fltrd_cutoff)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AAMDC</th>\n",
       "      <th>ABCA5</th>\n",
       "      <th>ABCG1</th>\n",
       "      <th>ACBD4</th>\n",
       "      <th>ACOX2</th>\n",
       "      <th>ACSF2</th>\n",
       "      <th>ADAMTS13</th>\n",
       "      <th>ADAMTSL5</th>\n",
       "      <th>ADCK5</th>\n",
       "      <th>AKR7A3</th>\n",
       "      <th>...</th>\n",
       "      <th>UBAC2</th>\n",
       "      <th>UPB1</th>\n",
       "      <th>VAMP2</th>\n",
       "      <th>VIPR1</th>\n",
       "      <th>WDTC1</th>\n",
       "      <th>YPEL3</th>\n",
       "      <th>YPEL5</th>\n",
       "      <th>ZFAND2B</th>\n",
       "      <th>ZFYVE19</th>\n",
       "      <th>ZFYVE28</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.8252</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.8021</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.8338</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.8149</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.8425</td>\n",
       "      <td>-0.8179</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 161 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    AAMDC  ABCA5  ABCG1   ACBD4  ACOX2  ACSF2  ADAMTS13  ADAMTSL5   ADCK5  \\\n",
       "5     0.0    0.0    0.0  0.0000    0.0    0.0       0.0       0.0  0.0000   \n",
       "8     0.0    0.0    0.0  0.0000    0.0    0.0       0.0       0.0  0.0000   \n",
       "17    0.0    0.0    0.0 -0.8252    0.0    0.0       0.0       0.0 -0.8021   \n",
       "19    0.0    0.0    0.0  0.0000    0.0    0.0       0.0       0.0  0.0000   \n",
       "36    0.0    0.0    0.0  0.0000    0.0    0.0       0.0       0.0  0.0000   \n",
       "\n",
       "    AKR7A3  ...  UBAC2  UPB1  VAMP2   VIPR1  WDTC1   YPEL3  YPEL5  ZFAND2B  \\\n",
       "5      0.0  ...    0.0   0.0    0.0  0.0000    0.0  0.0000    0.0   0.0000   \n",
       "8      0.0  ...    0.0   0.0    0.0  0.0000    0.0  0.0000    0.0   0.0000   \n",
       "17     0.0  ...    0.0   0.0    0.0 -0.8338    0.0 -0.8149    0.0  -0.8425   \n",
       "19     0.0  ...    0.0   0.0    0.0  0.0000    0.0  0.0000    0.0   0.0000   \n",
       "36     0.0  ...    0.0   0.0    0.0  0.0000    0.0  0.0000    0.0   0.0000   \n",
       "\n",
       "    ZFYVE19  ZFYVE28  \n",
       "5    0.0000      0.0  \n",
       "8    0.0000      0.0  \n",
       "17  -0.8179      0.0  \n",
       "19   0.0000      0.0  \n",
       "36   0.0000      0.0  \n",
       "\n",
       "[5 rows x 161 columns]"
      ]
     },
     "execution_count": 203,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "corr_fltrd_cutoff.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_metrics = compute_corr_metrics(corr_fltrd_cutoff, seq_feat, gene_map)\n",
    "corr_metrics.to_csv('/scratch/as3ek/histo_visual_recog/scripts/data/corr_metric_celiac_normal_cutoff_8.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_metrics = pd.read_csv('/scratch/as3ek/histo_visual_recog/scripts/data/corr_metric_celiac_normal_cutoff_7.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Feature</th>\n",
       "      <th>PearsonR</th>\n",
       "      <th>p-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>AAMDC</td>\n",
       "      <td>adipogenesis associated Mth938 domain containing</td>\n",
       "      <td>8</td>\n",
       "      <td>-0.737954</td>\n",
       "      <td>3.683384e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>AAMDC</td>\n",
       "      <td>adipogenesis associated Mth938 domain containing</td>\n",
       "      <td>15</td>\n",
       "      <td>-0.719808</td>\n",
       "      <td>1.133104e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>AAMDC</td>\n",
       "      <td>adipogenesis associated Mth938 domain containing</td>\n",
       "      <td>16</td>\n",
       "      <td>0.717032</td>\n",
       "      <td>1.335456e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>AAMDC</td>\n",
       "      <td>adipogenesis associated Mth938 domain containing</td>\n",
       "      <td>17</td>\n",
       "      <td>-0.734327</td>\n",
       "      <td>4.644255e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>AAMDC</td>\n",
       "      <td>adipogenesis associated Mth938 domain containing</td>\n",
       "      <td>21</td>\n",
       "      <td>-0.733887</td>\n",
       "      <td>4.775661e-08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Gene Symbol                                         Gene Name  Feature  \\\n",
       "0       AAMDC  adipogenesis associated Mth938 domain containing        8   \n",
       "1       AAMDC  adipogenesis associated Mth938 domain containing       15   \n",
       "2       AAMDC  adipogenesis associated Mth938 domain containing       16   \n",
       "3       AAMDC  adipogenesis associated Mth938 domain containing       17   \n",
       "4       AAMDC  adipogenesis associated Mth938 domain containing       21   \n",
       "\n",
       "   PearsonR       p-value  \n",
       "0 -0.737954  3.683384e-08  \n",
       "1 -0.719808  1.133104e-07  \n",
       "2  0.717032  1.335456e-07  \n",
       "3 -0.734327  4.644255e-08  \n",
       "4 -0.733887  4.775661e-08  "
      ]
     },
     "execution_count": 211,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "corr_metrics.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_metrics_grpd = corr_metrics[['Gene Symbol', 'Gene Name', 'Feature']].groupby(by='Gene Symbol', as_index=False).agg({'Gene Name': 'first',\n",
    "                                                                                                     'Feature': 'count'})\n",
    "\n",
    "corr_metrics_grpd.columns = ['Gene Symbol', 'Gene Name', 'Number of Features']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_metrics_grpd.to_csv('/scratch/as3ek/histo_visual_recog/scripts/data/corr_metric_grouped_celiac_normal_cutoff_7.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Number of Features</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>AAMDC</td>\n",
       "      <td>adipogenesis associated Mth938 domain containing</td>\n",
       "      <td>209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>AATK</td>\n",
       "      <td>apoptosis associated tyrosine kinase</td>\n",
       "      <td>134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ABAT</td>\n",
       "      <td>4-aminobutyrate aminotransferase</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ABCA10</td>\n",
       "      <td>ATP binding cassette subfamily A member 10</td>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>ABCA5</td>\n",
       "      <td>ATP binding cassette subfamily A member 5</td>\n",
       "      <td>142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1710</td>\n",
       "      <td>ZSCAN16</td>\n",
       "      <td>zinc finger and SCAN domain containing 16</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1711</td>\n",
       "      <td>ZSCAN26</td>\n",
       "      <td>zinc finger and SCAN domain containing 26</td>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1712</td>\n",
       "      <td>ZSCAN30</td>\n",
       "      <td>zinc finger and SCAN domain containing 30</td>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1713</td>\n",
       "      <td>ZSWIM5</td>\n",
       "      <td>zinc finger SWIM-type containing 5</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1714</td>\n",
       "      <td>ZSWIM8</td>\n",
       "      <td>zinc finger SWIM-type containing 8</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1715 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene Symbol                                         Gene Name  \\\n",
       "0          AAMDC  adipogenesis associated Mth938 domain containing   \n",
       "1           AATK              apoptosis associated tyrosine kinase   \n",
       "2           ABAT                  4-aminobutyrate aminotransferase   \n",
       "3         ABCA10        ATP binding cassette subfamily A member 10   \n",
       "4          ABCA5         ATP binding cassette subfamily A member 5   \n",
       "...          ...                                               ...   \n",
       "1710     ZSCAN16         zinc finger and SCAN domain containing 16   \n",
       "1711     ZSCAN26         zinc finger and SCAN domain containing 26   \n",
       "1712     ZSCAN30         zinc finger and SCAN domain containing 30   \n",
       "1713      ZSWIM5                zinc finger SWIM-type containing 5   \n",
       "1714      ZSWIM8                zinc finger SWIM-type containing 8   \n",
       "\n",
       "      Number of Features  \n",
       "0                    209  \n",
       "1                    134  \n",
       "2                      6  \n",
       "3                    120  \n",
       "4                    142  \n",
       "...                  ...  \n",
       "1710                  15  \n",
       "1711                  87  \n",
       "1712                 169  \n",
       "1713                   1  \n",
       "1714                   1  \n",
       "\n",
       "[1715 rows x 3 columns]"
      ]
     },
     "execution_count": 214,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "corr_metrics_grpd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.cluster import KMeans"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {},
   "outputs": [],
   "source": [
    "kmeans_all_data = KMeans(n_clusters=5, random_state=0).fit(corr_fltrd)\n",
    "kmeans_cutoff_data = KMeans(n_clusters=5, random_state=0).fit(df_cl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_fltrd['cluster'] = kmeans_all_data.labels_\n",
    "df_cl['cluster'] = kmeans_cutoff_data.labels_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {},
   "outputs": [],
   "source": [
    "feat_cluster = pd.DataFrame()\n",
    "\n",
    "for cluster in np.unique(df_cl['cluster']):\n",
    "    cl = df_cl[df_cl['cluster'] == cluster].drop('cluster', axis=1).copy().T\n",
    "    \n",
    "    # Remove all 0 values from columns\n",
    "    cl = cl.loc[:, (cl != 0).any(axis=0)]\n",
    "\n",
    "    # Remove all 0 values from rows\n",
    "    cl = cl[(cl.T != 0).any()]\n",
    "    \n",
    "    tmp = pd.DataFrame([cluster, list(cl.index), len(cl.index), list(cl.columns), len(cl.columns)]).T\n",
    "    feat_cluster = pd.concat([feat_cluster, tmp])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {},
   "outputs": [],
   "source": [
    "feat_cluster.columns = ['cluster', 'genes', 'num_genes', 'features', 'num_features']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {},
   "outputs": [],
   "source": [
    "feat_cluster.to_csv('/scratch/as3ek/histo_visual_recog/results/feature_clusters_cutoff_8.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-0.44"
      ]
     },
     "execution_count": 226,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.min(corr_fltrd['DUOXA2'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4035.613696"
      ]
     },
     "execution_count": 235,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.mean(normal_rna_seq[np.unique(sgnfct_seq_feat_corr['gene'])[0]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2174.0717784313724"
      ]
     },
     "execution_count": 234,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.mean(ee_rna_seq[np.unique(sgnfct_seq_feat_corr['gene'])[0]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['ALDOB', 'AMACR', 'APOH', 'CA2', 'CAPN13', 'CDK20', 'CHAD',\n",
       "       'CRIP3', 'CYP4F3', 'EPHX1', 'HGD', 'IGSF9', 'ISX', 'IYD', 'KCNK10',\n",
       "       'KIF12', 'LTK', 'MAPKAPK3', 'ME1', 'MMP24', 'MMP28', 'MOCS1',\n",
       "       'NELL2', 'OAT', 'PPP1R36', 'RAPGEFL1', 'RGS11', 'SEMA6C', 'SHMT1',\n",
       "       'SLC5A4', 'TMEM229A', 'TMEM25', 'TMEM52', 'UNC5D', 'UNC93A',\n",
       "       'UPB1', 'USP2'], dtype=object)"
      ]
     },
     "execution_count": 233,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.unique(sgnfct_seq_feat_corr['gene'])[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     3672.1885\n",
       "1     4264.6830\n",
       "2     3090.2253\n",
       "3     4540.7686\n",
       "4     3200.6025\n",
       "5     3714.6306\n",
       "6     5261.3057\n",
       "7     4697.8080\n",
       "8     6668.3076\n",
       "9     2550.5593\n",
       "10    4364.0550\n",
       "11    4898.0825\n",
       "12    6355.1100\n",
       "13    5055.7250\n",
       "14    3362.0535\n",
       "15    2191.2117\n",
       "16    2555.9316\n",
       "17    4323.9400\n",
       "18    4143.9870\n",
       "19    5283.7720\n",
       "20    3237.6394\n",
       "21    3434.0540\n",
       "22    3716.8413\n",
       "23    3360.2893\n",
       "24    2946.5710\n",
       "Name: ALDOB, dtype: float64"
      ]
     },
     "execution_count": 236,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_rna_seq[np.unique(sgnfct_seq_feat_corr['gene'])[0]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>A4GNT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAC</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>...</th>\n",
       "      <th>ZUFSP</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDB</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>0.3119</td>\n",
       "      <td>-0.3472</td>\n",
       "      <td>0.1445</td>\n",
       "      <td>-0.0967</td>\n",
       "      <td>0.3854</td>\n",
       "      <td>-0.1643</td>\n",
       "      <td>0.1294</td>\n",
       "      <td>-0.4289</td>\n",
       "      <td>-0.0932</td>\n",
       "      <td>0.1329</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.3946</td>\n",
       "      <td>0.1735</td>\n",
       "      <td>0.4780</td>\n",
       "      <td>0.5078</td>\n",
       "      <td>-0.2422</td>\n",
       "      <td>-0.4870</td>\n",
       "      <td>-0.2796</td>\n",
       "      <td>-0.1401</td>\n",
       "      <td>-0.5111</td>\n",
       "      <td>-0.3466</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>0.1284</td>\n",
       "      <td>-0.2042</td>\n",
       "      <td>0.1863</td>\n",
       "      <td>-0.3056</td>\n",
       "      <td>0.1060</td>\n",
       "      <td>-0.0096</td>\n",
       "      <td>0.2321</td>\n",
       "      <td>-0.2797</td>\n",
       "      <td>-0.2290</td>\n",
       "      <td>0.0615</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.3042</td>\n",
       "      <td>0.1258</td>\n",
       "      <td>0.2579</td>\n",
       "      <td>0.3757</td>\n",
       "      <td>-0.2778</td>\n",
       "      <td>-0.3434</td>\n",
       "      <td>-0.1957</td>\n",
       "      <td>0.0560</td>\n",
       "      <td>-0.3595</td>\n",
       "      <td>-0.2769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>-0.3313</td>\n",
       "      <td>0.3809</td>\n",
       "      <td>-0.1416</td>\n",
       "      <td>0.0465</td>\n",
       "      <td>-0.1911</td>\n",
       "      <td>-0.1501</td>\n",
       "      <td>-0.2423</td>\n",
       "      <td>0.4914</td>\n",
       "      <td>0.2693</td>\n",
       "      <td>-0.0293</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4153</td>\n",
       "      <td>-0.0648</td>\n",
       "      <td>-0.3472</td>\n",
       "      <td>-0.4914</td>\n",
       "      <td>0.3739</td>\n",
       "      <td>0.4142</td>\n",
       "      <td>0.3454</td>\n",
       "      <td>-0.2233</td>\n",
       "      <td>0.5010</td>\n",
       "      <td>0.3856</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>0.3825</td>\n",
       "      <td>-0.3396</td>\n",
       "      <td>0.1219</td>\n",
       "      <td>0.0234</td>\n",
       "      <td>0.2292</td>\n",
       "      <td>0.0570</td>\n",
       "      <td>0.2309</td>\n",
       "      <td>-0.4308</td>\n",
       "      <td>-0.2162</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.3335</td>\n",
       "      <td>0.1083</td>\n",
       "      <td>0.3519</td>\n",
       "      <td>0.3605</td>\n",
       "      <td>-0.1833</td>\n",
       "      <td>-0.2411</td>\n",
       "      <td>-0.2442</td>\n",
       "      <td>0.0422</td>\n",
       "      <td>-0.4327</td>\n",
       "      <td>-0.2243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>-0.1607</td>\n",
       "      <td>0.0719</td>\n",
       "      <td>0.1798</td>\n",
       "      <td>-0.2205</td>\n",
       "      <td>-0.0421</td>\n",
       "      <td>-0.0450</td>\n",
       "      <td>0.0937</td>\n",
       "      <td>0.0601</td>\n",
       "      <td>-0.0293</td>\n",
       "      <td>-0.0024</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0585</td>\n",
       "      <td>0.1563</td>\n",
       "      <td>0.1940</td>\n",
       "      <td>0.1574</td>\n",
       "      <td>0.1914</td>\n",
       "      <td>0.0768</td>\n",
       "      <td>0.1835</td>\n",
       "      <td>-0.0511</td>\n",
       "      <td>0.1600</td>\n",
       "      <td>0.1300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2043</td>\n",
       "      <td>0.2917</td>\n",
       "      <td>-0.4330</td>\n",
       "      <td>0.3311</td>\n",
       "      <td>-0.1984</td>\n",
       "      <td>0.3086</td>\n",
       "      <td>-0.0923</td>\n",
       "      <td>0.2482</td>\n",
       "      <td>-0.4771</td>\n",
       "      <td>-0.1667</td>\n",
       "      <td>0.0517</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.4201</td>\n",
       "      <td>0.2521</td>\n",
       "      <td>0.4859</td>\n",
       "      <td>0.5534</td>\n",
       "      <td>-0.1840</td>\n",
       "      <td>-0.4522</td>\n",
       "      <td>-0.1614</td>\n",
       "      <td>-0.0778</td>\n",
       "      <td>-0.4755</td>\n",
       "      <td>-0.2524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2044</td>\n",
       "      <td>0.0656</td>\n",
       "      <td>-0.2723</td>\n",
       "      <td>0.3559</td>\n",
       "      <td>-0.2524</td>\n",
       "      <td>0.0765</td>\n",
       "      <td>-0.0987</td>\n",
       "      <td>0.1507</td>\n",
       "      <td>-0.3535</td>\n",
       "      <td>-0.2127</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.3267</td>\n",
       "      <td>0.2501</td>\n",
       "      <td>0.3119</td>\n",
       "      <td>0.3026</td>\n",
       "      <td>-0.1424</td>\n",
       "      <td>-0.3544</td>\n",
       "      <td>-0.1184</td>\n",
       "      <td>-0.0337</td>\n",
       "      <td>-0.4055</td>\n",
       "      <td>-0.1784</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2045</td>\n",
       "      <td>0.1669</td>\n",
       "      <td>-0.3481</td>\n",
       "      <td>0.3248</td>\n",
       "      <td>-0.2280</td>\n",
       "      <td>0.1997</td>\n",
       "      <td>-0.0506</td>\n",
       "      <td>0.1406</td>\n",
       "      <td>-0.4249</td>\n",
       "      <td>-0.2032</td>\n",
       "      <td>0.0453</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.3808</td>\n",
       "      <td>0.2210</td>\n",
       "      <td>0.4234</td>\n",
       "      <td>0.4981</td>\n",
       "      <td>-0.2388</td>\n",
       "      <td>-0.4771</td>\n",
       "      <td>-0.1805</td>\n",
       "      <td>-0.0162</td>\n",
       "      <td>-0.4239</td>\n",
       "      <td>-0.2731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2046</td>\n",
       "      <td>-0.0881</td>\n",
       "      <td>0.2946</td>\n",
       "      <td>-0.3856</td>\n",
       "      <td>0.3001</td>\n",
       "      <td>-0.1771</td>\n",
       "      <td>0.1268</td>\n",
       "      <td>-0.1719</td>\n",
       "      <td>0.3417</td>\n",
       "      <td>0.1009</td>\n",
       "      <td>0.0299</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3229</td>\n",
       "      <td>-0.2371</td>\n",
       "      <td>-0.3443</td>\n",
       "      <td>-0.3795</td>\n",
       "      <td>0.1240</td>\n",
       "      <td>0.3586</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.0833</td>\n",
       "      <td>0.3025</td>\n",
       "      <td>0.1483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2047</td>\n",
       "      <td>0.3937</td>\n",
       "      <td>-0.3872</td>\n",
       "      <td>0.1710</td>\n",
       "      <td>-0.0626</td>\n",
       "      <td>0.4071</td>\n",
       "      <td>-0.0406</td>\n",
       "      <td>0.3663</td>\n",
       "      <td>-0.5345</td>\n",
       "      <td>-0.0093</td>\n",
       "      <td>0.2079</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.2998</td>\n",
       "      <td>0.2257</td>\n",
       "      <td>0.4830</td>\n",
       "      <td>0.5001</td>\n",
       "      <td>-0.1582</td>\n",
       "      <td>-0.3683</td>\n",
       "      <td>-0.1412</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>-0.4869</td>\n",
       "      <td>-0.2156</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2048 rows × 13464 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        A1BG    A1CF     A2M  A4GALT   A4GNT    AAAS    AACS   AADAC   AAED1  \\\n",
       "0     0.3119 -0.3472  0.1445 -0.0967  0.3854 -0.1643  0.1294 -0.4289 -0.0932   \n",
       "1     0.1284 -0.2042  0.1863 -0.3056  0.1060 -0.0096  0.2321 -0.2797 -0.2290   \n",
       "2    -0.3313  0.3809 -0.1416  0.0465 -0.1911 -0.1501 -0.2423  0.4914  0.2693   \n",
       "3     0.3825 -0.3396  0.1219  0.0234  0.2292  0.0570  0.2309 -0.4308 -0.2162   \n",
       "4    -0.1607  0.0719  0.1798 -0.2205 -0.0421 -0.0450  0.0937  0.0601 -0.0293   \n",
       "...      ...     ...     ...     ...     ...     ...     ...     ...     ...   \n",
       "2043  0.2917 -0.4330  0.3311 -0.1984  0.3086 -0.0923  0.2482 -0.4771 -0.1667   \n",
       "2044  0.0656 -0.2723  0.3559 -0.2524  0.0765 -0.0987  0.1507 -0.3535 -0.2127   \n",
       "2045  0.1669 -0.3481  0.3248 -0.2280  0.1997 -0.0506  0.1406 -0.4249 -0.2032   \n",
       "2046 -0.0881  0.2946 -0.3856  0.3001 -0.1771  0.1268 -0.1719  0.3417  0.1009   \n",
       "2047  0.3937 -0.3872  0.1710 -0.0626  0.4071 -0.0406  0.3663 -0.5345 -0.0093   \n",
       "\n",
       "       AAGAB  ...   ZUFSP    ZW10  ZWILCH   ZWINT    ZXDB    ZXDC  ZYG11B  \\\n",
       "0     0.1329  ... -0.3946  0.1735  0.4780  0.5078 -0.2422 -0.4870 -0.2796   \n",
       "1     0.0615  ... -0.3042  0.1258  0.2579  0.3757 -0.2778 -0.3434 -0.1957   \n",
       "2    -0.0293  ...  0.4153 -0.0648 -0.3472 -0.4914  0.3739  0.4142  0.3454   \n",
       "3     0.0134  ... -0.3335  0.1083  0.3519  0.3605 -0.1833 -0.2411 -0.2442   \n",
       "4    -0.0024  ... -0.0585  0.1563  0.1940  0.1574  0.1914  0.0768  0.1835   \n",
       "...      ...  ...     ...     ...     ...     ...     ...     ...     ...   \n",
       "2043  0.0517  ... -0.4201  0.2521  0.4859  0.5534 -0.1840 -0.4522 -0.1614   \n",
       "2044 -0.0640  ... -0.3267  0.2501  0.3119  0.3026 -0.1424 -0.3544 -0.1184   \n",
       "2045  0.0453  ... -0.3808  0.2210  0.4234  0.4981 -0.2388 -0.4771 -0.1805   \n",
       "2046  0.0299  ...  0.3229 -0.2371 -0.3443 -0.3795  0.1240  0.3586  0.0411   \n",
       "2047  0.2079  ... -0.2998  0.2257  0.4830  0.5001 -0.1582 -0.3683 -0.1412   \n",
       "\n",
       "         ZYX   ZZEF1    ZZZ3  \n",
       "0    -0.1401 -0.5111 -0.3466  \n",
       "1     0.0560 -0.3595 -0.2769  \n",
       "2    -0.2233  0.5010  0.3856  \n",
       "3     0.0422 -0.4327 -0.2243  \n",
       "4    -0.0511  0.1600  0.1300  \n",
       "...      ...     ...     ...  \n",
       "2043 -0.0778 -0.4755 -0.2524  \n",
       "2044 -0.0337 -0.4055 -0.1784  \n",
       "2045 -0.0162 -0.4239 -0.2731  \n",
       "2046  0.0833  0.3025  0.1483  \n",
       "2047  0.0005 -0.4869 -0.2156  \n",
       "\n",
       "[2048 rows x 13464 columns]"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "corr_fltrd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
