{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "from scipy.stats.stats import pearsonr\n",
    "import pingouin as pg\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load features from the CNN model\n",
    "train_feat_max = pd.read_csv('/project/DSone/as3ek/data/csv_files/seem_train_2048_features_max_normal_all__ee_vsi.csv')\n",
    "valid_feat_max = pd.read_csv('/project/DSone/as3ek/data/csv_files/seem_valid_2048_features_max_normal_all__ee_vsi.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load cleaned EE RNA Seq Data\n",
    "ee_rna_seq = pd.read_csv('/project/DSone/as3ek/data/csv_files/ee_trascript.csv')\n",
    "ee_rna_seq = ee_rna_seq.drop(['Unnamed: 0'], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/apps/software/standard/core/anaconda/5.2.0-py3.6/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (13468,13469,13470) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "# Load RNA Seq Data\n",
    "rna_seq = pd.read_csv('/project/DSone/as3ek/data/csv_files/transcriptome_merged.csv')\n",
    "\n",
    "# Drop unnecessary columns\n",
    "rna_seq = rna_seq.drop(['Gene', 'PID', 'RNASes_Bx_loc', 'SEEM ID', 'Bx Date', 'Accession #'], axis=1)\n",
    "\n",
    "# Filter all data to get only normal RNA Seq data\n",
    "normal_rna_seq = rna_seq[rna_seq['Diagnosis'] == 'CONTROL'].reset_index(drop=True)\n",
    "normal_rna_seq = normal_rna_seq.drop('Diagnosis', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>A4GNT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAC</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>...</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDB</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "      <th>FileNames</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>2.938328</td>\n",
       "      <td>19.990509</td>\n",
       "      <td>83.735050</td>\n",
       "      <td>2.656216</td>\n",
       "      <td>3.404470</td>\n",
       "      <td>17.141710</td>\n",
       "      <td>11.889671</td>\n",
       "      <td>57.604313</td>\n",
       "      <td>2.052005</td>\n",
       "      <td>22.352093</td>\n",
       "      <td>...</td>\n",
       "      <td>7.167301</td>\n",
       "      <td>6.760921</td>\n",
       "      <td>23.934557</td>\n",
       "      <td>1.351300</td>\n",
       "      <td>5.770310</td>\n",
       "      <td>2.624760</td>\n",
       "      <td>32.301320</td>\n",
       "      <td>16.912931</td>\n",
       "      <td>10.427124</td>\n",
       "      <td>SEEM_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>4.698266</td>\n",
       "      <td>15.687117</td>\n",
       "      <td>92.621060</td>\n",
       "      <td>1.957990</td>\n",
       "      <td>0.306087</td>\n",
       "      <td>12.404878</td>\n",
       "      <td>9.844436</td>\n",
       "      <td>84.911210</td>\n",
       "      <td>1.641775</td>\n",
       "      <td>21.663334</td>\n",
       "      <td>...</td>\n",
       "      <td>7.521921</td>\n",
       "      <td>6.561717</td>\n",
       "      <td>18.160603</td>\n",
       "      <td>1.151140</td>\n",
       "      <td>6.536171</td>\n",
       "      <td>3.283770</td>\n",
       "      <td>34.461420</td>\n",
       "      <td>14.909936</td>\n",
       "      <td>13.230409</td>\n",
       "      <td>SEEM_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>3.455630</td>\n",
       "      <td>13.785192</td>\n",
       "      <td>107.816505</td>\n",
       "      <td>3.281972</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>13.406025</td>\n",
       "      <td>8.643127</td>\n",
       "      <td>49.870205</td>\n",
       "      <td>2.645186</td>\n",
       "      <td>18.025269</td>\n",
       "      <td>...</td>\n",
       "      <td>4.720486</td>\n",
       "      <td>4.125326</td>\n",
       "      <td>13.924426</td>\n",
       "      <td>0.878572</td>\n",
       "      <td>4.932373</td>\n",
       "      <td>2.385005</td>\n",
       "      <td>29.004587</td>\n",
       "      <td>15.764729</td>\n",
       "      <td>6.827882</td>\n",
       "      <td>SEEM_4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>2.466141</td>\n",
       "      <td>36.609490</td>\n",
       "      <td>135.728450</td>\n",
       "      <td>3.121654</td>\n",
       "      <td>0.293090</td>\n",
       "      <td>17.032896</td>\n",
       "      <td>12.018597</td>\n",
       "      <td>91.219810</td>\n",
       "      <td>3.087415</td>\n",
       "      <td>23.395113</td>\n",
       "      <td>...</td>\n",
       "      <td>8.026385</td>\n",
       "      <td>7.405385</td>\n",
       "      <td>14.101945</td>\n",
       "      <td>2.055490</td>\n",
       "      <td>7.218950</td>\n",
       "      <td>4.769273</td>\n",
       "      <td>30.454320</td>\n",
       "      <td>25.666950</td>\n",
       "      <td>14.545485</td>\n",
       "      <td>SEEM_5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>3.249304</td>\n",
       "      <td>30.674992</td>\n",
       "      <td>115.543710</td>\n",
       "      <td>2.711289</td>\n",
       "      <td>1.881430</td>\n",
       "      <td>19.385916</td>\n",
       "      <td>11.410261</td>\n",
       "      <td>96.130104</td>\n",
       "      <td>1.817835</td>\n",
       "      <td>18.924387</td>\n",
       "      <td>...</td>\n",
       "      <td>6.723240</td>\n",
       "      <td>6.889716</td>\n",
       "      <td>18.950125</td>\n",
       "      <td>1.843850</td>\n",
       "      <td>7.449580</td>\n",
       "      <td>3.398000</td>\n",
       "      <td>22.919546</td>\n",
       "      <td>26.630081</td>\n",
       "      <td>12.240758</td>\n",
       "      <td>SEEM_6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 13465 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       A1BG       A1CF         A2M    A4GALT     A4GNT       AAAS       AACS  \\\n",
       "0  2.938328  19.990509   83.735050  2.656216  3.404470  17.141710  11.889671   \n",
       "1  4.698266  15.687117   92.621060  1.957990  0.306087  12.404878   9.844436   \n",
       "2  3.455630  13.785192  107.816505  3.281972  0.010000  13.406025   8.643127   \n",
       "3  2.466141  36.609490  135.728450  3.121654  0.293090  17.032896  12.018597   \n",
       "4  3.249304  30.674992  115.543710  2.711289  1.881430  19.385916  11.410261   \n",
       "\n",
       "       AADAC     AAED1      AAGAB  ...      ZW10    ZWILCH      ZWINT  \\\n",
       "0  57.604313  2.052005  22.352093  ...  7.167301  6.760921  23.934557   \n",
       "1  84.911210  1.641775  21.663334  ...  7.521921  6.561717  18.160603   \n",
       "2  49.870205  2.645186  18.025269  ...  4.720486  4.125326  13.924426   \n",
       "3  91.219810  3.087415  23.395113  ...  8.026385  7.405385  14.101945   \n",
       "4  96.130104  1.817835  18.924387  ...  6.723240  6.889716  18.950125   \n",
       "\n",
       "       ZXDB      ZXDC    ZYG11B        ZYX      ZZEF1       ZZZ3  FileNames  \n",
       "0  1.351300  5.770310  2.624760  32.301320  16.912931  10.427124     SEEM_1  \n",
       "1  1.151140  6.536171  3.283770  34.461420  14.909936  13.230409     SEEM_3  \n",
       "2  0.878572  4.932373  2.385005  29.004587  15.764729   6.827882     SEEM_4  \n",
       "3  2.055490  7.218950  4.769273  30.454320  25.666950  14.545485     SEEM_5  \n",
       "4  1.843850  7.449580  3.398000  22.919546  26.630081  12.240758     SEEM_6  \n",
       "\n",
       "[5 rows x 13465 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ee_rna_seq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>A4GNT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAC</th>\n",
       "      <th>AAED1</th>\n",
       "      <th>AAGAB</th>\n",
       "      <th>...</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDB</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "      <th>FileNames</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1.527489</td>\n",
       "      <td>39.161606</td>\n",
       "      <td>141.393680</td>\n",
       "      <td>2.250741</td>\n",
       "      <td>3.340980</td>\n",
       "      <td>13.393648</td>\n",
       "      <td>11.521561</td>\n",
       "      <td>190.26294</td>\n",
       "      <td>3.802879</td>\n",
       "      <td>27.817423</td>\n",
       "      <td>...</td>\n",
       "      <td>7.966211</td>\n",
       "      <td>5.824094</td>\n",
       "      <td>15.380531</td>\n",
       "      <td>1.90331</td>\n",
       "      <td>6.084198</td>\n",
       "      <td>5.986877</td>\n",
       "      <td>19.513960</td>\n",
       "      <td>30.423355</td>\n",
       "      <td>13.065509</td>\n",
       "      <td>GI-17-2781A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1.836630</td>\n",
       "      <td>20.815561</td>\n",
       "      <td>70.397380</td>\n",
       "      <td>1.366245</td>\n",
       "      <td>8.056910</td>\n",
       "      <td>13.729294</td>\n",
       "      <td>11.792672</td>\n",
       "      <td>188.86170</td>\n",
       "      <td>2.597826</td>\n",
       "      <td>19.777740</td>\n",
       "      <td>...</td>\n",
       "      <td>6.067741</td>\n",
       "      <td>3.992341</td>\n",
       "      <td>12.340939</td>\n",
       "      <td>1.04346</td>\n",
       "      <td>4.722662</td>\n",
       "      <td>2.653590</td>\n",
       "      <td>24.902603</td>\n",
       "      <td>14.547653</td>\n",
       "      <td>10.382214</td>\n",
       "      <td>GI-17-5415A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>2.070108</td>\n",
       "      <td>30.049267</td>\n",
       "      <td>78.129480</td>\n",
       "      <td>2.200778</td>\n",
       "      <td>8.668561</td>\n",
       "      <td>14.197790</td>\n",
       "      <td>13.160649</td>\n",
       "      <td>299.86722</td>\n",
       "      <td>2.576309</td>\n",
       "      <td>22.687897</td>\n",
       "      <td>...</td>\n",
       "      <td>6.487394</td>\n",
       "      <td>4.760917</td>\n",
       "      <td>12.637662</td>\n",
       "      <td>1.09973</td>\n",
       "      <td>5.131794</td>\n",
       "      <td>4.623030</td>\n",
       "      <td>23.566916</td>\n",
       "      <td>17.086746</td>\n",
       "      <td>8.982910</td>\n",
       "      <td>GI-17-5671A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>2.323137</td>\n",
       "      <td>27.274525</td>\n",
       "      <td>93.843880</td>\n",
       "      <td>2.508640</td>\n",
       "      <td>1.156290</td>\n",
       "      <td>15.076123</td>\n",
       "      <td>11.050514</td>\n",
       "      <td>176.36838</td>\n",
       "      <td>1.392020</td>\n",
       "      <td>19.399889</td>\n",
       "      <td>...</td>\n",
       "      <td>4.988570</td>\n",
       "      <td>2.749776</td>\n",
       "      <td>13.213606</td>\n",
       "      <td>1.35546</td>\n",
       "      <td>5.087205</td>\n",
       "      <td>2.442820</td>\n",
       "      <td>26.665813</td>\n",
       "      <td>20.346804</td>\n",
       "      <td>7.571774</td>\n",
       "      <td>GI-17-6710A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1.550304</td>\n",
       "      <td>28.912386</td>\n",
       "      <td>78.544624</td>\n",
       "      <td>2.113191</td>\n",
       "      <td>0.260323</td>\n",
       "      <td>14.713420</td>\n",
       "      <td>10.754949</td>\n",
       "      <td>272.44060</td>\n",
       "      <td>2.326984</td>\n",
       "      <td>22.267065</td>\n",
       "      <td>...</td>\n",
       "      <td>7.437720</td>\n",
       "      <td>5.820460</td>\n",
       "      <td>18.061235</td>\n",
       "      <td>1.39190</td>\n",
       "      <td>5.067470</td>\n",
       "      <td>3.432585</td>\n",
       "      <td>24.128992</td>\n",
       "      <td>15.404562</td>\n",
       "      <td>10.151729</td>\n",
       "      <td>GI-17-6728A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 13465 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       A1BG       A1CF         A2M    A4GALT     A4GNT       AAAS       AACS  \\\n",
       "0  1.527489  39.161606  141.393680  2.250741  3.340980  13.393648  11.521561   \n",
       "1  1.836630  20.815561   70.397380  1.366245  8.056910  13.729294  11.792672   \n",
       "2  2.070108  30.049267   78.129480  2.200778  8.668561  14.197790  13.160649   \n",
       "3  2.323137  27.274525   93.843880  2.508640  1.156290  15.076123  11.050514   \n",
       "4  1.550304  28.912386   78.544624  2.113191  0.260323  14.713420  10.754949   \n",
       "\n",
       "       AADAC     AAED1      AAGAB  ...      ZW10    ZWILCH      ZWINT  \\\n",
       "0  190.26294  3.802879  27.817423  ...  7.966211  5.824094  15.380531   \n",
       "1  188.86170  2.597826  19.777740  ...  6.067741  3.992341  12.340939   \n",
       "2  299.86722  2.576309  22.687897  ...  6.487394  4.760917  12.637662   \n",
       "3  176.36838  1.392020  19.399889  ...  4.988570  2.749776  13.213606   \n",
       "4  272.44060  2.326984  22.267065  ...  7.437720  5.820460  18.061235   \n",
       "\n",
       "      ZXDB      ZXDC    ZYG11B        ZYX      ZZEF1       ZZZ3    FileNames  \n",
       "0  1.90331  6.084198  5.986877  19.513960  30.423355  13.065509  GI-17-2781A  \n",
       "1  1.04346  4.722662  2.653590  24.902603  14.547653  10.382214  GI-17-5415A  \n",
       "2  1.09973  5.131794  4.623030  23.566916  17.086746   8.982910  GI-17-5671A  \n",
       "3  1.35546  5.087205  2.442820  26.665813  20.346804   7.571774  GI-17-6710A  \n",
       "4  1.39190  5.067470  3.432585  24.128992  15.404562  10.151729  GI-17-6728A  \n",
       "\n",
       "[5 rows x 13465 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_rna_seq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Concat and save clean RNA seq data\n",
    "rna_seq_clean = pd.concat([ee_rna_seq, normal_rna_seq], ignore_index=True)\n",
    "rna_seq_clean.to_csv('/project/DSone/as3ek/data/csv_files/rna_seq_clean_normal_ee.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter features to get EE data and merge from train and valid\n",
    "ee_train_feat_max = train_feat_max[train_feat_max['fname'].str.contains('SEEM')].reset_index(drop=True)\n",
    "ee_valid_feat_max = valid_feat_max[valid_feat_max['fname'].str.contains('SEEM')].reset_index(drop=True)\n",
    "ee_feat_max = pd.concat([ee_train_feat_max, ee_valid_feat_max])\n",
    "\n",
    "# Clean the name field\n",
    "ee_feat_max['fname'] = ee_feat_max['fname'].str.split('__').str[0].str.split('_').str[0] + '_' + ee_feat_max['fname'].str.split('__').str[0].str.split('_').str[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter features to get Normal data and merge from train and valid\n",
    "normal_train_feat_max = train_feat_max[train_feat_max['fname'].str.contains('G')].reset_index(drop=True)\n",
    "normal_valid_feat_max = valid_feat_max[valid_feat_max['fname'].str.contains('G')].reset_index(drop=True)\n",
    "normal_feat_max = pd.concat([normal_train_feat_max, normal_valid_feat_max], ignore_index=True)\n",
    "\n",
    "# Clean the name field\n",
    "normal_feat_max['fname'] = normal_feat_max['fname'].str.split('_').str[0].str.replace('GI', 'GI-').str.split(' ').str[0] + 'A'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Group features for all patches by biopsy name\n",
    "ee_feat_max_grpd = ee_feat_max.groupby('fname', as_index=False).mean()\n",
    "normal_feat_max_grpd = normal_feat_max.groupby('fname', as_index=False).mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fname</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>...</th>\n",
       "      <th>2038</th>\n",
       "      <th>2039</th>\n",
       "      <th>2040</th>\n",
       "      <th>2041</th>\n",
       "      <th>2042</th>\n",
       "      <th>2043</th>\n",
       "      <th>2044</th>\n",
       "      <th>2045</th>\n",
       "      <th>2046</th>\n",
       "      <th>2047</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>SEEM_1</td>\n",
       "      <td>1.400954</td>\n",
       "      <td>2.726676</td>\n",
       "      <td>1.868812</td>\n",
       "      <td>1.917903</td>\n",
       "      <td>2.655619</td>\n",
       "      <td>2.859661</td>\n",
       "      <td>2.283474</td>\n",
       "      <td>2.000574</td>\n",
       "      <td>1.911550</td>\n",
       "      <td>...</td>\n",
       "      <td>1.892482</td>\n",
       "      <td>1.881015</td>\n",
       "      <td>1.763218</td>\n",
       "      <td>1.135075</td>\n",
       "      <td>3.023325</td>\n",
       "      <td>0.631242</td>\n",
       "      <td>2.081944</td>\n",
       "      <td>1.175545</td>\n",
       "      <td>1.302622</td>\n",
       "      <td>1.588411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>SEEM_10</td>\n",
       "      <td>1.347156</td>\n",
       "      <td>3.181994</td>\n",
       "      <td>2.297989</td>\n",
       "      <td>1.938909</td>\n",
       "      <td>2.182216</td>\n",
       "      <td>2.793844</td>\n",
       "      <td>2.065651</td>\n",
       "      <td>2.069440</td>\n",
       "      <td>1.613876</td>\n",
       "      <td>...</td>\n",
       "      <td>2.755671</td>\n",
       "      <td>2.465041</td>\n",
       "      <td>2.863062</td>\n",
       "      <td>1.014355</td>\n",
       "      <td>3.783387</td>\n",
       "      <td>0.726808</td>\n",
       "      <td>2.891178</td>\n",
       "      <td>1.032845</td>\n",
       "      <td>1.699797</td>\n",
       "      <td>2.078044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>SEEM_11</td>\n",
       "      <td>1.093749</td>\n",
       "      <td>3.349344</td>\n",
       "      <td>3.042262</td>\n",
       "      <td>1.958739</td>\n",
       "      <td>1.869790</td>\n",
       "      <td>2.074053</td>\n",
       "      <td>1.838703</td>\n",
       "      <td>2.658418</td>\n",
       "      <td>1.740939</td>\n",
       "      <td>...</td>\n",
       "      <td>3.729062</td>\n",
       "      <td>3.444447</td>\n",
       "      <td>2.597348</td>\n",
       "      <td>0.715201</td>\n",
       "      <td>4.651663</td>\n",
       "      <td>0.684788</td>\n",
       "      <td>3.045861</td>\n",
       "      <td>0.770555</td>\n",
       "      <td>1.404238</td>\n",
       "      <td>3.087810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>SEEM_12</td>\n",
       "      <td>1.034657</td>\n",
       "      <td>3.299655</td>\n",
       "      <td>2.598511</td>\n",
       "      <td>2.353304</td>\n",
       "      <td>1.631794</td>\n",
       "      <td>2.352409</td>\n",
       "      <td>1.752420</td>\n",
       "      <td>2.765432</td>\n",
       "      <td>1.396725</td>\n",
       "      <td>...</td>\n",
       "      <td>3.285949</td>\n",
       "      <td>2.576968</td>\n",
       "      <td>3.049833</td>\n",
       "      <td>0.748103</td>\n",
       "      <td>3.937245</td>\n",
       "      <td>0.978531</td>\n",
       "      <td>2.349350</td>\n",
       "      <td>0.777661</td>\n",
       "      <td>1.627161</td>\n",
       "      <td>2.904910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>SEEM_13</td>\n",
       "      <td>1.146579</td>\n",
       "      <td>2.898013</td>\n",
       "      <td>2.440671</td>\n",
       "      <td>2.175549</td>\n",
       "      <td>2.432416</td>\n",
       "      <td>2.872315</td>\n",
       "      <td>1.321863</td>\n",
       "      <td>2.363730</td>\n",
       "      <td>2.061244</td>\n",
       "      <td>...</td>\n",
       "      <td>3.272165</td>\n",
       "      <td>2.882773</td>\n",
       "      <td>3.017845</td>\n",
       "      <td>0.867513</td>\n",
       "      <td>4.961912</td>\n",
       "      <td>0.632133</td>\n",
       "      <td>3.110673</td>\n",
       "      <td>0.687398</td>\n",
       "      <td>2.304506</td>\n",
       "      <td>2.189863</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 2049 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     fname         0         1         2         3         4         5  \\\n",
       "0   SEEM_1  1.400954  2.726676  1.868812  1.917903  2.655619  2.859661   \n",
       "1  SEEM_10  1.347156  3.181994  2.297989  1.938909  2.182216  2.793844   \n",
       "2  SEEM_11  1.093749  3.349344  3.042262  1.958739  1.869790  2.074053   \n",
       "3  SEEM_12  1.034657  3.299655  2.598511  2.353304  1.631794  2.352409   \n",
       "4  SEEM_13  1.146579  2.898013  2.440671  2.175549  2.432416  2.872315   \n",
       "\n",
       "          6         7         8  ...      2038      2039      2040      2041  \\\n",
       "0  2.283474  2.000574  1.911550  ...  1.892482  1.881015  1.763218  1.135075   \n",
       "1  2.065651  2.069440  1.613876  ...  2.755671  2.465041  2.863062  1.014355   \n",
       "2  1.838703  2.658418  1.740939  ...  3.729062  3.444447  2.597348  0.715201   \n",
       "3  1.752420  2.765432  1.396725  ...  3.285949  2.576968  3.049833  0.748103   \n",
       "4  1.321863  2.363730  2.061244  ...  3.272165  2.882773  3.017845  0.867513   \n",
       "\n",
       "       2042      2043      2044      2045      2046      2047  \n",
       "0  3.023325  0.631242  2.081944  1.175545  1.302622  1.588411  \n",
       "1  3.783387  0.726808  2.891178  1.032845  1.699797  2.078044  \n",
       "2  4.651663  0.684788  3.045861  0.770555  1.404238  3.087810  \n",
       "3  3.937245  0.978531  2.349350  0.777661  1.627161  2.904910  \n",
       "4  4.961912  0.632133  3.110673  0.687398  2.304506  2.189863  \n",
       "\n",
       "[5 rows x 2049 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ee_feat_max_grpd.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fname</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>...</th>\n",
       "      <th>2038</th>\n",
       "      <th>2039</th>\n",
       "      <th>2040</th>\n",
       "      <th>2041</th>\n",
       "      <th>2042</th>\n",
       "      <th>2043</th>\n",
       "      <th>2044</th>\n",
       "      <th>2045</th>\n",
       "      <th>2046</th>\n",
       "      <th>2047</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>GI-17-10289A</td>\n",
       "      <td>1.705304</td>\n",
       "      <td>1.308385</td>\n",
       "      <td>1.214186</td>\n",
       "      <td>1.144616</td>\n",
       "      <td>2.019404</td>\n",
       "      <td>1.173569</td>\n",
       "      <td>1.481161</td>\n",
       "      <td>1.411003</td>\n",
       "      <td>1.604710</td>\n",
       "      <td>...</td>\n",
       "      <td>1.353803</td>\n",
       "      <td>1.501212</td>\n",
       "      <td>1.755425</td>\n",
       "      <td>2.285431</td>\n",
       "      <td>1.236964</td>\n",
       "      <td>1.350781</td>\n",
       "      <td>1.407560</td>\n",
       "      <td>1.979489</td>\n",
       "      <td>3.193440</td>\n",
       "      <td>1.053154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>GI-17-10498A</td>\n",
       "      <td>1.791951</td>\n",
       "      <td>1.338543</td>\n",
       "      <td>1.063339</td>\n",
       "      <td>0.861534</td>\n",
       "      <td>2.173868</td>\n",
       "      <td>1.278980</td>\n",
       "      <td>0.839176</td>\n",
       "      <td>1.369282</td>\n",
       "      <td>2.094078</td>\n",
       "      <td>...</td>\n",
       "      <td>1.764162</td>\n",
       "      <td>1.548570</td>\n",
       "      <td>1.801140</td>\n",
       "      <td>2.439009</td>\n",
       "      <td>1.137720</td>\n",
       "      <td>1.467445</td>\n",
       "      <td>1.001710</td>\n",
       "      <td>2.256842</td>\n",
       "      <td>3.805646</td>\n",
       "      <td>0.982195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>GI-17-10706A</td>\n",
       "      <td>1.624928</td>\n",
       "      <td>1.510126</td>\n",
       "      <td>1.291649</td>\n",
       "      <td>1.120004</td>\n",
       "      <td>2.066083</td>\n",
       "      <td>1.305926</td>\n",
       "      <td>1.127223</td>\n",
       "      <td>1.419618</td>\n",
       "      <td>1.683323</td>\n",
       "      <td>...</td>\n",
       "      <td>1.777002</td>\n",
       "      <td>1.732200</td>\n",
       "      <td>1.997007</td>\n",
       "      <td>2.087791</td>\n",
       "      <td>1.456396</td>\n",
       "      <td>1.212907</td>\n",
       "      <td>1.362405</td>\n",
       "      <td>1.793345</td>\n",
       "      <td>3.366465</td>\n",
       "      <td>1.193909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>GI-17-2781A</td>\n",
       "      <td>1.737263</td>\n",
       "      <td>1.149561</td>\n",
       "      <td>1.115752</td>\n",
       "      <td>0.711680</td>\n",
       "      <td>1.691020</td>\n",
       "      <td>0.998992</td>\n",
       "      <td>1.301657</td>\n",
       "      <td>1.142493</td>\n",
       "      <td>1.653945</td>\n",
       "      <td>...</td>\n",
       "      <td>1.295423</td>\n",
       "      <td>1.388237</td>\n",
       "      <td>1.726299</td>\n",
       "      <td>2.544281</td>\n",
       "      <td>1.158321</td>\n",
       "      <td>1.524205</td>\n",
       "      <td>1.003201</td>\n",
       "      <td>2.495913</td>\n",
       "      <td>3.502311</td>\n",
       "      <td>0.951937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>GI-17-5073A</td>\n",
       "      <td>1.639812</td>\n",
       "      <td>1.273617</td>\n",
       "      <td>1.188617</td>\n",
       "      <td>0.867676</td>\n",
       "      <td>2.264301</td>\n",
       "      <td>1.224853</td>\n",
       "      <td>0.857378</td>\n",
       "      <td>1.247174</td>\n",
       "      <td>2.027124</td>\n",
       "      <td>...</td>\n",
       "      <td>1.428486</td>\n",
       "      <td>1.408774</td>\n",
       "      <td>1.793765</td>\n",
       "      <td>2.605442</td>\n",
       "      <td>1.279758</td>\n",
       "      <td>1.315401</td>\n",
       "      <td>1.038976</td>\n",
       "      <td>2.253473</td>\n",
       "      <td>3.944400</td>\n",
       "      <td>0.843002</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 2049 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          fname         0         1         2         3         4         5  \\\n",
       "0  GI-17-10289A  1.705304  1.308385  1.214186  1.144616  2.019404  1.173569   \n",
       "1  GI-17-10498A  1.791951  1.338543  1.063339  0.861534  2.173868  1.278980   \n",
       "2  GI-17-10706A  1.624928  1.510126  1.291649  1.120004  2.066083  1.305926   \n",
       "3   GI-17-2781A  1.737263  1.149561  1.115752  0.711680  1.691020  0.998992   \n",
       "4   GI-17-5073A  1.639812  1.273617  1.188617  0.867676  2.264301  1.224853   \n",
       "\n",
       "          6         7         8  ...      2038      2039      2040      2041  \\\n",
       "0  1.481161  1.411003  1.604710  ...  1.353803  1.501212  1.755425  2.285431   \n",
       "1  0.839176  1.369282  2.094078  ...  1.764162  1.548570  1.801140  2.439009   \n",
       "2  1.127223  1.419618  1.683323  ...  1.777002  1.732200  1.997007  2.087791   \n",
       "3  1.301657  1.142493  1.653945  ...  1.295423  1.388237  1.726299  2.544281   \n",
       "4  0.857378  1.247174  2.027124  ...  1.428486  1.408774  1.793765  2.605442   \n",
       "\n",
       "       2042      2043      2044      2045      2046      2047  \n",
       "0  1.236964  1.350781  1.407560  1.979489  3.193440  1.053154  \n",
       "1  1.137720  1.467445  1.001710  2.256842  3.805646  0.982195  \n",
       "2  1.456396  1.212907  1.362405  1.793345  3.366465  1.193909  \n",
       "3  1.158321  1.524205  1.003201  2.495913  3.502311  0.951937  \n",
       "4  1.279758  1.315401  1.038976  2.253473  3.944400  0.843002  \n",
       "\n",
       "[5 rows x 2049 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_feat_max_grpd.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "feat_max_grpd_clean = pd.concat([ee_feat_max_grpd, normal_feat_max_grpd], ignore_index=True)\n",
    "feat_max_grpd_clean.to_csv('/scratch/as3ek/histo_visual_recog/scripts/data/max_features_grouped_clean_SEEM_GI.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(46, 15514)"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ee_seq_feat.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(14, 15514)"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_seq_feat.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge seq with features\n",
    "ee_seq_feat = pd.merge(ee_rna_seq, ee_feat_max_grpd, left_on='FileNames', right_on='fname')\n",
    "normal_seq_feat = pd.merge(normal_rna_seq, normal_feat_max_grpd, left_on='FileNames', right_on='fname')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Concate EE and normal frames\n",
    "seq_feat = pd.concat([ee_seq_feat, normal_seq_feat], ignore_index=True)\n",
    "seq_feat.to_csv('/scratch/as3ek/histo_visual_recog/results/rnaseq_feat.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Getting list of all columns for correlation b/w\n",
    "x = list(ee_rna_seq)\n",
    "x.remove('FileNames')\n",
    "\n",
    "y = list(ee_feat_max_grpd)\n",
    "y.remove('fname')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Calculate pairwise correlation and filter it for reqd columns and round the values\n",
    "corr = seq_feat.corr()\n",
    "corr_fltrd = corr[x][corr.index.isin(y)]\n",
    "corr_fltrd = corr_fltrd.round(4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_fltrd.to_csv('/scratch/as3ek/histo_visual_recog/results/gene_feat_corr.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_map = pd.read_csv('/scratch/as3ek/histo_visual_recog/scripts/data/custom.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "def corr_cutoff(corr_fltrd, cutoff):\n",
    "    # Make copy\n",
    "    df = corr_fltrd.copy().T\n",
    "    \n",
    "    # Replace all values below cutoff with 0\n",
    "    for col in y:\n",
    "        df.loc[np.abs(df[col]) < cutoff, col] = 0\n",
    "    \n",
    "    # Remove all columns with all zeros (Remove unnecessary features)\n",
    "    df = df.loc[:, (df != 0).any(axis=0)]\n",
    "    \n",
    "    # Transpose and remove all columns with all zeros (Remove unnecessary genes)\n",
    "    df = df.T\n",
    "    df = df.loc[:, (df != 0).any(axis=0)]\n",
    "    \n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "def compute_corr_metrics(corr_fltrd_cutoff, seq_feat, gene_map):\n",
    "    corr_metrics = pd.DataFrame(columns=['Gene Symbol', 'Gene Name', 'Feature', 'PearsonR', 'p-value'])\n",
    "    c = corr_fltrd_cutoff.copy()\n",
    "    for gene in c.columns:\n",
    "        df = pd.DataFrame(c[gene]).T\n",
    "        df = df.loc[:, (df != 0).any(axis=0)]\n",
    "        \n",
    "        names = list(gene_map[gene_map['Approved symbol'] == gene]['Approved name'])\n",
    "        if len(names) > 0:\n",
    "            name = names[0]\n",
    "        else:\n",
    "            name = gene\n",
    "        \n",
    "        for feat in df.columns:\n",
    "            r, p = pearsonr(seq_feat[feat], seq_feat[gene])\n",
    "            tmp = pd.DataFrame([[gene, name, feat, r, p]], columns=['Gene Symbol', 'Gene Name', 'Feature', 'PearsonR', 'p-value'])\n",
    "            corr_metrics = pd.concat([corr_metrics, tmp], ignore_index=True)\n",
    "    return corr_metrics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "def compute_regulation(normal_rna_seq, ee_rna_seq, genes, cutoff_fold, gene_map):\n",
    "    regulation_data = pd.DataFrame(columns=['Regulation', 'Gene Symbol', 'Gene Name', 'fold'])\n",
    "    for col in genes:\n",
    "        a = np.mean(ee_rna_seq[col]) / np.mean(normal_rna_seq[col]) \n",
    "        b = 1 / a\n",
    "        names = list(gene_map[gene_map['Approved symbol'] == col]['Approved name'])\n",
    "        if len(names) > 0:\n",
    "            name = names[0]\n",
    "        else:\n",
    "            name = col\n",
    "        if np.abs(a) > cutoff_fold:\n",
    "            tmp = pd.DataFrame([['up', col, name, a]], columns=['Regulation', 'Gene Symbol', 'Gene Name', 'fold'])\n",
    "            regulation_data = pd.concat([regulation_data, tmp], ignore_index=True)\n",
    "        if np.abs(b) > cutoff_fold:\n",
    "            tmp = pd.DataFrame([['down', col, name, b]], columns=['Regulation', 'Gene Symbol', 'Gene Name', 'fold'])\n",
    "            regulation_data = pd.concat([regulation_data, tmp], ignore_index=True)\n",
    "    \n",
    "    return regulation_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [],
   "source": [
    "regulation_data = compute_regulation(normal_rna_seq, ee_rna_seq, x, 1, gene_map)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "regulation_data.to_csv('/scratch/as3ek/histo_visual_recog/scripts/data/gene_regulation_data.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "corr_fltrd_cutoff = corr_cutoff(corr_fltrd, 0.6)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "plt.figure(figsize=(30, 200))\n",
    "sns.heatmap(corr_fltrd_cutoff)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_metrics = compute_corr_metrics(corr_fltrd_cutoff, seq_feat, gene_map)\n",
    "corr_metrics.to_csv('/scratch/as3ek/histo_visual_recog/scripts/data/corr_metric_cutoff_6.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.cluster import KMeans"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {},
   "outputs": [],
   "source": [
    "kmeans_all_data = KMeans(n_clusters=5, random_state=0).fit(corr_fltrd)\n",
    "kmeans_cutoff_data = KMeans(n_clusters=5, random_state=0).fit(df_cl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_fltrd['cluster'] = kmeans_all_data.labels_\n",
    "df_cl['cluster'] = kmeans_cutoff_data.labels_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {},
   "outputs": [],
   "source": [
    "feat_cluster = pd.DataFrame()\n",
    "\n",
    "for cluster in np.unique(df_cl['cluster']):\n",
    "    cl = df_cl[df_cl['cluster'] == cluster].drop('cluster', axis=1).copy().T\n",
    "    \n",
    "    # Remove all 0 values from columns\n",
    "    cl = cl.loc[:, (cl != 0).any(axis=0)]\n",
    "\n",
    "    # Remove all 0 values from rows\n",
    "    cl = cl[(cl.T != 0).any()]\n",
    "    \n",
    "    tmp = pd.DataFrame([cluster, list(cl.index), len(cl.index), list(cl.columns), len(cl.columns)]).T\n",
    "    feat_cluster = pd.concat([feat_cluster, tmp])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {},
   "outputs": [],
   "source": [
    "feat_cluster.columns = ['cluster', 'genes', 'num_genes', 'features', 'num_features']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {},
   "outputs": [],
   "source": [
    "feat_cluster.to_csv('/scratch/as3ek/histo_visual_recog/results/feature_clusters_cutoff_8.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-0.44"
      ]
     },
     "execution_count": 226,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.min(corr_fltrd['DUOXA2'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4035.613696"
      ]
     },
     "execution_count": 235,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.mean(normal_rna_seq[np.unique(sgnfct_seq_feat_corr['gene'])[0]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2174.0717784313724"
      ]
     },
     "execution_count": 234,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.mean(ee_rna_seq[np.unique(sgnfct_seq_feat_corr['gene'])[0]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['ALDOB', 'AMACR', 'APOH', 'CA2', 'CAPN13', 'CDK20', 'CHAD',\n",
       "       'CRIP3', 'CYP4F3', 'EPHX1', 'HGD', 'IGSF9', 'ISX', 'IYD', 'KCNK10',\n",
       "       'KIF12', 'LTK', 'MAPKAPK3', 'ME1', 'MMP24', 'MMP28', 'MOCS1',\n",
       "       'NELL2', 'OAT', 'PPP1R36', 'RAPGEFL1', 'RGS11', 'SEMA6C', 'SHMT1',\n",
       "       'SLC5A4', 'TMEM229A', 'TMEM25', 'TMEM52', 'UNC5D', 'UNC93A',\n",
       "       'UPB1', 'USP2'], dtype=object)"
      ]
     },
     "execution_count": 233,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.unique(sgnfct_seq_feat_corr['gene'])[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     3672.1885\n",
       "1     4264.6830\n",
       "2     3090.2253\n",
       "3     4540.7686\n",
       "4     3200.6025\n",
       "5     3714.6306\n",
       "6     5261.3057\n",
       "7     4697.8080\n",
       "8     6668.3076\n",
       "9     2550.5593\n",
       "10    4364.0550\n",
       "11    4898.0825\n",
       "12    6355.1100\n",
       "13    5055.7250\n",
       "14    3362.0535\n",
       "15    2191.2117\n",
       "16    2555.9316\n",
       "17    4323.9400\n",
       "18    4143.9870\n",
       "19    5283.7720\n",
       "20    3237.6394\n",
       "21    3434.0540\n",
       "22    3716.8413\n",
       "23    3360.2893\n",
       "24    2946.5710\n",
       "Name: ALDOB, dtype: float64"
      ]
     },
     "execution_count": 236,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_rna_seq[np.unique(sgnfct_seq_feat_corr['gene'])[0]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
